<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Basic checks of BioPlex PPI data • BioPlex</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Basic checks of BioPlex PPI data">
<meta property="og:description" content="BioPlex">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BioPlex</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.5.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/BioPlex.html">Data retrieval</a>
</li>
<li>
  <a href="../articles/BasicChecks.html">Data checks</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ccb-hms/BioPlex" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Basic checks of BioPlex PPI data</h1>
            
            <h4 data-toc-skip class="date">20 February 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ccb-hms/BioPlex/blob/HEAD/vignettes/BasicChecks.Rmd" class="external-link"><code>vignettes/BasicChecks.Rmd</code></a></small>
      <div class="hidden name"><code>BasicChecks.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ccb-hms/BioPlex" class="external-link">BioPlex</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/AnnotationDbi" class="external-link">AnnotationDbi</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">graph</span><span class="op">)</span></code></pre></div>
<p>Connect to <a href="http://bioconductor.org/packages/AnnotationHub" class="external-link">AnnotationHub</a>:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ah</span> <span class="op">&lt;-</span> <span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>Connect to <a href="http://bioconductor.org/packages/ExperimentHub" class="external-link">ExperimentHub</a>:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu">ExperimentHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>OrgDb package for human:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orgdb</span> <span class="op">&lt;-</span> <span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"orgDb"</span>, <span class="st">"Homo sapiens"</span><span class="op">)</span><span class="op">)</span>
<span class="va">orgdb</span> <span class="op">&lt;-</span> <span class="va">orgdb</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="va">orgdb</span>
<span class="co">#&gt; OrgDb object:</span>
<span class="co">#&gt; | DBSCHEMAVERSION: 2.1</span>
<span class="co">#&gt; | Db type: OrgDb</span>
<span class="co">#&gt; | Supporting package: AnnotationDbi</span>
<span class="co">#&gt; | DBSCHEMA: HUMAN_DB</span>
<span class="co">#&gt; | ORGANISM: Homo sapiens</span>
<span class="co">#&gt; | SPECIES: Human</span>
<span class="co">#&gt; | EGSOURCEDATE: 2022-Sep12</span>
<span class="co">#&gt; | EGSOURCENAME: Entrez Gene</span>
<span class="co">#&gt; | EGSOURCEURL: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA</span>
<span class="co">#&gt; | CENTRALID: EG</span>
<span class="co">#&gt; | TAXID: 9606</span>
<span class="co">#&gt; | GOSOURCENAME: Gene Ontology</span>
<span class="co">#&gt; | GOSOURCEURL: http://current.geneontology.org/ontology/go-basic.obo</span>
<span class="co">#&gt; | GOSOURCEDATE: 2022-07-01</span>
<span class="co">#&gt; | GOEGSOURCEDATE: 2022-Sep12</span>
<span class="co">#&gt; | GOEGSOURCENAME: Entrez Gene</span>
<span class="co">#&gt; | GOEGSOURCEURL: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA</span>
<span class="co">#&gt; | KEGGSOURCENAME: KEGG GENOME</span>
<span class="co">#&gt; | KEGGSOURCEURL: ftp://ftp.genome.jp/pub/kegg/genomes</span>
<span class="co">#&gt; | KEGGSOURCEDATE: 2011-Mar15</span>
<span class="co">#&gt; | GPSOURCENAME: UCSC Genome Bioinformatics (Homo sapiens)</span>
<span class="co">#&gt; | GPSOURCEURL: </span>
<span class="co">#&gt; | GPSOURCEDATE: 2022-Aug31</span>
<span class="co">#&gt; | ENSOURCEDATE: 2022-Jun28</span>
<span class="co">#&gt; | ENSOURCENAME: Ensembl</span>
<span class="co">#&gt; | ENSOURCEURL: ftp://ftp.ensembl.org/pub/current_fasta</span>
<span class="co">#&gt; | UPSOURCENAME: Uniprot</span>
<span class="co">#&gt; | UPSOURCEURL: http://www.UniProt.org/</span>
<span class="co">#&gt; | UPSOURCEDATE: Fri Sep 23 16:26:35 2022</span>
<span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">keytypes</a></span><span class="op">(</span><span class="va">orgdb</span><span class="op">)</span>
<span class="co">#&gt;  [1] "ACCNUM"       "ALIAS"        "ENSEMBL"      "ENSEMBLPROT"  "ENSEMBLTRANS"</span>
<span class="co">#&gt;  [6] "ENTREZID"     "ENZYME"       "EVIDENCE"     "EVIDENCEALL"  "GENENAME"    </span>
<span class="co">#&gt; [11] "GENETYPE"     "GO"           "GOALL"        "IPI"          "MAP"         </span>
<span class="co">#&gt; [16] "OMIM"         "ONTOLOGY"     "ONTOLOGYALL"  "PATH"         "PFAM"        </span>
<span class="co">#&gt; [21] "PMID"         "PROSITE"      "REFSEQ"       "SYMBOL"       "UCSCKG"      </span>
<span class="co">#&gt; [26] "UNIPROT"</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="check-identify-corum-complexes-that-have-a-subunit-of-interest">Check: identify CORUM complexes that have a subunit of interest<a class="anchor" aria-label="anchor" href="#check-identify-corum-complexes-that-have-a-subunit-of-interest"></a>
</h2>
<p>Get core set of complexes:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">core</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getCorum.html">getCorum</a></span><span class="op">(</span>set <span class="op">=</span> <span class="st">"core"</span>, organism <span class="op">=</span> <span class="st">"Human"</span><span class="op">)</span>
<span class="co">#&gt; Using cached version from 2023-02-20 10:50:59</span></code></pre></div>
<p>Turn the CORUM complexes into a list of graph instances, where all
nodes of a complex are connected to all other nodes of that complex with
undirected edges.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">core.glist</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/corum2graphlist.html">corum2graphlist</a></span><span class="op">(</span><span class="va">core</span>, subunit.id.type <span class="op">=</span> <span class="st">"UNIPROT"</span><span class="op">)</span></code></pre></div>
<p>Identify complexes that have a subunit of interest:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">has.cdk2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hasSubunit.html">hasSubunit</a></span><span class="op">(</span><span class="va">core.glist</span>, 
                       subunit <span class="op">=</span> <span class="st">"CDK2"</span>,
                       id.type <span class="op">=</span> <span class="st">"SYMBOL"</span><span class="op">)</span></code></pre></div>
<p>Check the answer:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">has.cdk2</span><span class="op">)</span>
<span class="co">#&gt; has.cdk2</span>
<span class="co">#&gt; FALSE  TRUE </span>
<span class="co">#&gt;  2408     9</span>
<span class="va">cdk2.glist</span> <span class="op">&lt;-</span> <span class="va">core.glist</span><span class="op">[</span><span class="va">has.cdk2</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">cdk2.glist</span>, <span class="kw">function</span><span class="op">(</span><span class="va">g</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/nodeData-methods.html" class="external-link">nodeData</a></span><span class="op">(</span><span class="va">g</span>, attr <span class="op">=</span> <span class="st">"SYMBOL"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; $CORUM311_Cell_cycle_kinase_complex_CDK2</span>
<span class="co">#&gt;   P12004   P24385   P24941   P38936 </span>
<span class="co">#&gt;   "PCNA"  "CCND1"   "CDK2" "CDKN1A" </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`CORUM1003_RC_complex_(Replication_competent_complex)`</span>
<span class="co">#&gt;  P09884  P20248  P24941  P35249  P35250  P35251  P40937  P40938  Q14181 </span>
<span class="co">#&gt; "POLA1" "CCNA2"  "CDK2"  "RFC4"  "RFC2"  "RFC1"  "RFC5"  "RFC3" "POLA2" </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`CORUM1004_RC_complex_during_S-phase_of_cell_cycle`</span>
<span class="co">#&gt;  P09874  P09884  P11387  P15927  P18858  P20248  P24941  P27694  P28340  P35244 </span>
<span class="co">#&gt; "PARP1" "POLA1"  "TOP1"  "RPA2"  "LIG1" "CCNA2"  "CDK2"  "RPA1" "POLD1"  "RPA3" </span>
<span class="co">#&gt;  P35250  P35251  Q07864 </span>
<span class="co">#&gt;  "RFC2"  "RFC1"  "POLE" </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`CORUM1656_p27-cyclinE-CDK2_complex`</span>
<span class="co">#&gt;   P24864   P24941   P46527 </span>
<span class="co">#&gt;  "CCNE1"   "CDK2" "CDKN1B" </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`CORUM3015_p27-cyclinE-Cdk2_-_Ubiquitin_E3_ligase_(SKP1A,_SKP2,_CUL1,_CKS1B,_RBX1)_complex`</span>
<span class="co">#&gt;   P24864   P24941   P46527   P61024   P62877   P63208   Q13309   Q13616 </span>
<span class="co">#&gt;  "CCNE1"   "CDK2" "CDKN1B"  "CKS1B"   "RBX1"   "SKP1"   "SKP2"   "CUL1" </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`CORUM5556_CDK2-CCNA2_complex`</span>
<span class="co">#&gt;  P20248  P24941 </span>
<span class="co">#&gt; "CCNA2"  "CDK2" </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`CORUM5559_CDC2-CCNA2-CDK2_complex`</span>
<span class="co">#&gt;  P06493  P20248  P24941 </span>
<span class="co">#&gt;  "CDK1" "CCNA2"  "CDK2" </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`CORUM5560_CDK2-CCNE1_complex`</span>
<span class="co">#&gt;  P24864  P24941 </span>
<span class="co">#&gt; "CCNE1"  "CDK2" </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`CORUM6589_E2F-1-DP-1-cyclinA-CDK2_complex`</span>
<span class="co">#&gt;  P24941  P78396  Q01094  Q14186 </span>
<span class="co">#&gt;  "CDK2" "CCNA1"  "E2F1" "TFDP1"</span></code></pre></div>
<p>We can then also inspect the graph with plotting utilities from the
<a href="https://bioconductor.org/packages/Rgraphviz" class="external-link">Rgraphviz</a>
package:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">cdk2.glist</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, main <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cdk2.glist</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="check-extract-bioplex-ppis-for-a-corum-complex">Check: extract BioPlex PPIs for a CORUM complex<a class="anchor" aria-label="anchor" href="#check-extract-bioplex-ppis-for-a-corum-complex"></a>
</h2>
<p>Get the latest version of the 293T PPI network:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.293t</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getBioPlex.html">getBioPlex</a></span><span class="op">(</span>cell.line <span class="op">=</span> <span class="st">"293T"</span>, version <span class="op">=</span> <span class="st">"3.0"</span><span class="op">)</span>
<span class="co">#&gt; Using cached version from 2023-02-20 10:51:04</span></code></pre></div>
<p>Turn the BioPlex PPI network into one big graph where bait and prey
relationship are represented by directed edges from bait to prey.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bioplex2graph.html">bioplex2graph</a></span><span class="op">(</span><span class="va">bp.293t</span><span class="op">)</span></code></pre></div>
<p>Now we can also easily pull out a BioPlex subnetwork for a CORUM
complex of interest:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">n</span> <span class="op">&lt;-</span> <span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/graphNEL-class.html" class="external-link">nodes</a></span><span class="op">(</span><span class="va">cdk2.glist</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="va">bp.sgr</span> <span class="op">&lt;-</span> <span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/subGraph.html" class="external-link">subGraph</a></span><span class="op">(</span><span class="va">n</span>, <span class="va">bp.gr</span><span class="op">)</span>
<span class="va">bp.sgr</span>
<span class="co">#&gt; A graphNEL graph with directed edges</span>
<span class="co">#&gt; Number of Nodes = 4 </span>
<span class="co">#&gt; Number of Edges = 5</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="check-identify-interacting-domains-for-a-pfam-domain-of-interest">Check: identify interacting domains for a PFAM domain of
interest<a class="anchor" aria-label="anchor" href="#check-identify-interacting-domains-for-a-pfam-domain-of-interest"></a>
</h2>
<p>Add PFAM domain annotations to the node metadata:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.gr</span> <span class="op">&lt;-</span> <span class="fu">BioPlex</span><span class="fu">::</span><span class="fu"><a href="../reference/annotatePFAM.html">annotatePFAM</a></span><span class="op">(</span><span class="va">bp.gr</span>, <span class="va">orgdb</span><span class="op">)</span></code></pre></div>
<p>Create a map from PFAM to UNIPROT:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">unip2pfam</span> <span class="op">&lt;-</span> <span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/nodeData-methods.html" class="external-link">nodeData</a></span><span class="op">(</span><span class="va">bp.gr</span>, <span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/graphNEL-class.html" class="external-link">nodes</a></span><span class="op">(</span><span class="va">bp.gr</span><span class="op">)</span>, <span class="st">"PFAM"</span><span class="op">)</span>
<span class="va">pfam2unip</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/stack.html" class="external-link">stack</a></span><span class="op">(</span><span class="va">unip2pfam</span><span class="op">)</span>
<span class="va">pfam2unip</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">pfam2unip</span><span class="op">$</span><span class="va">ind</span><span class="op">)</span>, <span class="va">pfam2unip</span><span class="op">$</span><span class="va">values</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pfam2unip</span>, <span class="fl">2</span><span class="op">)</span>
<span class="co">#&gt; $PF00001</span>
<span class="co">#&gt;  [1] "P28566" "P25106" "P23945" "Q9HBX9" "P16473" "P04201" "Q9HC97" "P30968"</span>
<span class="co">#&gt;  [9] "Q9Y2T6" "Q14330" "P46089" "Q15391" "Q9BXA5" "Q13304" "P61073" "P21462"</span>
<span class="co">#&gt; [17] "P25090" "Q99679" "P21730" "P30556" "P43088" "P32246" "P32249" "Q9Y2T5"</span>
<span class="co">#&gt; [25] "Q7Z602" "P43657" "O00398" "Q9H244" "Q86VZ1" "Q9NPB9" "Q99788" "P51684"</span>
<span class="co">#&gt; [33] "P35414" "O00590" "Q9H1Y3" "P55085" "O15218" "Q9GZQ4" "P25101" "Q9NS66"</span>
<span class="co">#&gt; [41] "Q9NQS5" "P21453" "P14416" "P24530" "P32239" "Q16581" "O00421" "Q9UHM6"</span>
<span class="co">#&gt; [49] "Q8N6U8" "P20309" "O15354" "Q9BXC0" "P47775" "P30550" "P49146" "P47900"</span>
<span class="co">#&gt; [57] "Q8TDU9" "P25103" "P35372" "P41597" "Q9P296" "P28335" "O95136" "P08173"</span>
<span class="co">#&gt; [65] "P29371" "P41146" "P43119" "O95977" "Q9HBW0" "Q99677" "Q9BXB1" "Q8WXD0"</span>
<span class="co">#&gt; [73] "O43193" "P30989" "Q8NGU9" "P47901" "P22888" "Q9GZN0" "P21917" "O60755"</span>
<span class="co">#&gt; [81] "Q8TDV0" "O43614" "Q9NS67" "P08912" "Q9UPC5" "Q8TDV2" "Q92633" "Q9NQ55"</span>
<span class="co">#&gt; [89] "Q13585" "Q9UBY5" "Q9H228" "P28222"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $PF00002</span>
<span class="co">#&gt;  [1] "Q8IZP9" "P41587" "Q8IZF4" "P49190" "P32241" "P47871" "P48960" "Q8IZF5"</span>
<span class="co">#&gt;  [9] "O14514" "Q03431" "Q9NYQ6" "Q9HCU4" "Q8WXG9" "Q9NYQ7" "O60242" "O60241"</span>
<span class="co">#&gt; [17] "Q9HAR2" "O94910" "Q8IWK6" "O95490" "Q96PE1" "Q86SQ4"</span></code></pre></div>
<p>Let’s focus on <a href="http://pfam.xfam.org/family/PF02023" class="external-link">PF02023</a>, corresponding to
the zinc finger-associated SCAN domain. For each protein containing the
SCAN domain, we now extract PFAM domains connected to the SCAN domain by
an edge in the BioPlex network.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">scan.unip</span> <span class="op">&lt;-</span> <span class="va">pfam2unip</span><span class="op">[[</span><span class="st">"PF02023"</span><span class="op">]</span><span class="op">]</span>
<span class="va">getIAPfams</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/nodeData-methods.html" class="external-link">nodeData</a></span><span class="op">(</span><span class="va">bp.gr</span>, <span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/graphNEL-class.html" class="external-link">edges</a></span><span class="op">(</span><span class="va">bp.gr</span><span class="op">)</span><span class="op">[[</span><span class="va">n</span><span class="op">]</span><span class="op">]</span>, <span class="st">"PFAM"</span><span class="op">)</span>
<span class="va">unip2iapfams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">scan.unip</span>, <span class="va">getIAPfams</span><span class="op">)</span>
<span class="va">unip2iapfams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">unip2iapfams</span>, <span class="va">unlist</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">unip2iapfams</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">scan.unip</span></code></pre></div>
<p>Looking at the top 5 PFAM domains most frequently connected to the
SCAN domain by an edge in the BioPlex network …</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pfam2iapfams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">unip2iapfams</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">pfam2iapfams</span><span class="op">)</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>
<span class="co">#&gt; pfam2iapfams</span>
<span class="co">#&gt; PF02023 PF00096 PF01352 PF06467 PF00249 </span>
<span class="co">#&gt;     208     169     100      14       8</span></code></pre></div>
<p>… we find <a href="http://pfam.xfam.org/family/PF02023" class="external-link">PF02023</a>,
the SCAN domain itself, and <a href="http://pfam.xfam.org/family/PF00096" class="external-link">PF00096</a>, a C2H2 type zinc
finger domain. This finding is consistent with results reported in the
<a href="https://doi.org/10.1016/j.cell.2021.04.011" class="external-link">BioPlex 3.0
publication</a>.</p>
<p>See also the <a href="https://ccb-hms.github.io/BioPlexAnalysis/articles/PFAM.html">PFAM
domain-domain association analysis vignette</a> for a more comprehensive
analysis of PFAM domain associations in the BioPlex network.</p>
</div>
<div class="section level2">
<h2 id="check-agreement-between-cnv-tracks">Check: agreement between CNV tracks<a class="anchor" aria-label="anchor" href="#check-agreement-between-cnv-tracks"></a>
</h2>
<p>Genomic data from whole-genome sequencing for six different lineages
of the human embryonic kidney HEK293 cell line can be obtained from <a href="http://hek293genome.org" class="external-link">hek293genome.org</a>.</p>
<p>This includes copy number variation (CNV) data for the 293T cell
line. Available CNV tracks include (i) CNV regions inferred from
sequencing read-depth analysis, and (ii) CNV regions inferred from
Illumina SNP arrays.</p>
<p>Here, we check agreement between inferred copy numbers from both
assay types.</p>
<p>We start by obtaining genomic coordinates and copy number scores from
the sequencing track …</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cnv.hmm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getHEK293GenomeTrack.html">getHEK293GenomeTrack</a></span><span class="op">(</span>track <span class="op">=</span> <span class="st">"cnv.hmm"</span>, cell.line <span class="op">=</span> <span class="st">"293T"</span><span class="op">)</span>
<span class="co">#&gt; Using cached version from 2023-02-20 10:51:13</span>
<span class="va">cnv.hmm</span>
<span class="co">#&gt; GRanges object with 12382 ranges and 1 metadata column:</span>
<span class="co">#&gt;           seqnames              ranges strand |     score</span>
<span class="co">#&gt;              &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;</span>
<span class="co">#&gt;       [1]     chr1       823231-829231      * |      3.26</span>
<span class="co">#&gt;       [2]     chr1       835231-913231      * |      3.08</span>
<span class="co">#&gt;       [3]     chr1      923231-1063231      * |      3.20</span>
<span class="co">#&gt;       [4]     chr1     1079231-1213231      * |      3.21</span>
<span class="co">#&gt;       [5]     chr1     1223231-1399231      * |      3.27</span>
<span class="co">#&gt;       ...      ...                 ...    ... .       ...</span>
<span class="co">#&gt;   [12378]     chrX 154750237-154762237      * |      3.96</span>
<span class="co">#&gt;   [12379]     chrX 154778237-154780237      * |      4.02</span>
<span class="co">#&gt;   [12380]     chrX 154802237-154822237      * |      3.79</span>
<span class="co">#&gt;   [12381]     chrX 154842237-154846237      * |      3.85</span>
<span class="co">#&gt;   [12382]     chrM             0-12000      * |      8.82</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 24 sequences from hg18 genome; no seqlengths</span></code></pre></div>
<p>… and from the SNP track.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cnv.snp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getHEK293GenomeTrack.html">getHEK293GenomeTrack</a></span><span class="op">(</span>track <span class="op">=</span> <span class="st">"cnv.snp"</span>, cell.line <span class="op">=</span> <span class="st">"293T"</span><span class="op">)</span>
<span class="co">#&gt; Using cached version from 2023-02-20 10:51:15</span>
<span class="va">cnv.snp</span>
<span class="co">#&gt; GRanges object with 204 ranges and 1 metadata column:</span>
<span class="co">#&gt;         seqnames              ranges strand |     score</span>
<span class="co">#&gt;            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="co">#&gt;     [1]     chr1      742429-5117504      * |         3</span>
<span class="co">#&gt;     [2]     chr1    5126548-27691288      * |         4</span>
<span class="co">#&gt;     [3]     chr1   27696717-33934376      * |         6</span>
<span class="co">#&gt;     [4]     chr1   33946560-48944728      * |         4</span>
<span class="co">#&gt;     [5]     chr1   48956914-50523337      * |         2</span>
<span class="co">#&gt;     ...      ...                 ...    ... .       ...</span>
<span class="co">#&gt;   [200]     chrX 106780442-127925246      * |         5</span>
<span class="co">#&gt;   [201]     chrX 127931857-131443550      * |         6</span>
<span class="co">#&gt;   [202]     chrX 131541633-136116260      * |         5</span>
<span class="co">#&gt;   [203]     chrX 136125141-147982191      * |         4</span>
<span class="co">#&gt;   [204]     chrX 147983995-154582606      * |         5</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 23 sequences from hg18 genome; no seqlengths</span></code></pre></div>
<p>We reduce the check for agreement between both CNV tracks by
transferring copy numbers to overlapping genes, and subsequently, assess
the agreement between the resulting gene copy numbers for both
tracks.</p>
<p>As the genomic coordinates from both CNV tracks is based on the hg18
human genome assembly, we obtain gene coordinates for hg18 from
AnnotationHub:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TxDb"</span>, <span class="st">"Homo sapiens"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; AnnotationHub with 39 records</span>
<span class="co">#&gt; # snapshotDate(): 2022-10-31</span>
<span class="co">#&gt; # $dataprovider: GENCODE, UCSC, NCBI, tRNAdb, snoRNAdb, RMBase v2.0</span>
<span class="co">#&gt; # $species: Homo sapiens</span>
<span class="co">#&gt; # $rdataclass: TxDb, SQLiteFile, ChainFile, FaFile</span>
<span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span>
<span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span>
<span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span>
<span class="co">#&gt; # retrieve records with, e.g., 'object[["AH52256"]]' </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;              title                                             </span>
<span class="co">#&gt;   AH52256  | TxDb.Hsapiens.BioMart.igis.sqlite                 </span>
<span class="co">#&gt;   AH52257  | TxDb.Hsapiens.UCSC.hg18.knownGene.sqlite          </span>
<span class="co">#&gt;   AH52258  | TxDb.Hsapiens.UCSC.hg19.knownGene.sqlite          </span>
<span class="co">#&gt;   AH52259  | TxDb.Hsapiens.UCSC.hg19.lincRNAsTranscripts.sqlite</span>
<span class="co">#&gt;   AH52260  | TxDb.Hsapiens.UCSC.hg38.knownGene.sqlite          </span>
<span class="co">#&gt;   ...        ...                                               </span>
<span class="co">#&gt;   AH92592  | TxDb.Hsapiens.UCSC.hg38.refGene.sqlite            </span>
<span class="co">#&gt;   AH97949  | TxDb.Hsapiens.UCSC.hg38.knownGene.sqlite          </span>
<span class="co">#&gt;   AH100418 | TxDb.Hsapiens.UCSC.hg38.knownGene.sqlite          </span>
<span class="co">#&gt;   AH100419 | TxDb.Hsapiens.UCSC.hg38.refGene.sqlite            </span>
<span class="co">#&gt;   AH107068 | TxDb.Hsapiens.UCSC.hg38.knownGene.sqlite</span>
<span class="va">txdb</span> <span class="op">&lt;-</span> <span class="va">ah</span><span class="op">[[</span><span class="st">"AH52257"</span><span class="op">]</span><span class="op">]</span>
<span class="co">#&gt; loading from cache</span>
<span class="co">#&gt; Loading required package: GenomicFeatures</span>
<span class="va">gs</span> <span class="op">&lt;-</span> <span class="fu">GenomicFeatures</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">txdb</span><span class="op">)</span>
<span class="co">#&gt;   379 genes were dropped because they have exons located on both strands</span>
<span class="co">#&gt;   of the same reference sequence or on more than one reference sequence,</span>
<span class="co">#&gt;   so cannot be represented by a single genomic range.</span>
<span class="co">#&gt;   Use 'single.strand.genes.only=FALSE' to get all the genes in a</span>
<span class="co">#&gt;   GRangesList object, or use suppressMessages() to suppress this message.</span>
<span class="va">gs</span>
<span class="co">#&gt; GRanges object with 19742 ranges and 1 metadata column:</span>
<span class="co">#&gt;         seqnames              ranges strand |     gene_id</span>
<span class="co">#&gt;            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">#&gt;       1    chr19   63549984-63565932      - |           1</span>
<span class="co">#&gt;      10     chr8   18293035-18303003      + |          10</span>
<span class="co">#&gt;     100    chr20   42681577-42713790      - |         100</span>
<span class="co">#&gt;    1000    chr18   23784933-24011189      - |        1000</span>
<span class="co">#&gt;   10000     chr1 241718158-242073207      - |       10000</span>
<span class="co">#&gt;     ...      ...                 ...    ... .         ...</span>
<span class="co">#&gt;    9991     chr9 114020538-114135733      - |        9991</span>
<span class="co">#&gt;    9992    chr21   34658193-34665310      + |        9992</span>
<span class="co">#&gt;    9993    chr22   17403795-17489967      - |        9993</span>
<span class="co">#&gt;    9994     chr6   90596334-90640876      + |        9994</span>
<span class="co">#&gt;    9997    chr22   49308863-49310900      - |        9997</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 49 sequences (1 circular) from hg18 genome</span></code></pre></div>
<p>We then transfer SNP-inferred copy numbers to genes by overlap …</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">olaps</span> <span class="op">&lt;-</span> <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps</a></span><span class="op">(</span><span class="va">gs</span>, <span class="va">cnv.snp</span><span class="op">)</span>
<span class="va">joined</span> <span class="op">&lt;-</span> <span class="va">gs</span><span class="op">[</span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">queryHits</a></span><span class="op">(</span><span class="va">olaps</span><span class="op">)</span><span class="op">]</span>
<span class="va">joined</span><span class="op">$</span><span class="va">score</span> <span class="op">&lt;-</span> <span class="va">cnv.snp</span><span class="op">$</span><span class="va">score</span><span class="op">[</span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">subjectHits</a></span><span class="op">(</span><span class="va">olaps</span><span class="op">)</span><span class="op">]</span>
<span class="va">joined</span>
<span class="co">#&gt; GRanges object with 19555 ranges and 2 metadata columns:</span>
<span class="co">#&gt;         seqnames              ranges strand |     gene_id     score</span>
<span class="co">#&gt;            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;integer&gt;</span>
<span class="co">#&gt;       1    chr19   63549984-63565932      - |           1         4</span>
<span class="co">#&gt;      10     chr8   18293035-18303003      + |          10         2</span>
<span class="co">#&gt;     100    chr20   42681577-42713790      - |         100         4</span>
<span class="co">#&gt;    1000    chr18   23784933-24011189      - |        1000         3</span>
<span class="co">#&gt;   10000     chr1 241718158-242073207      - |       10000         6</span>
<span class="co">#&gt;     ...      ...                 ...    ... .         ...       ...</span>
<span class="co">#&gt;    9991     chr9 114020538-114135733      - |        9991         3</span>
<span class="co">#&gt;    9992    chr21   34658193-34665310      + |        9992         4</span>
<span class="co">#&gt;    9993    chr22   17403795-17489967      - |        9993         5</span>
<span class="co">#&gt;    9994     chr6   90596334-90640876      + |        9994         3</span>
<span class="co">#&gt;    9997    chr22   49308863-49310900      - |        9997         4</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 49 sequences (1 circular) from hg18 genome</span></code></pre></div>
<p>… and, analogously, transfer sequencing-inferred copy numbers to
genes by overlap.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">olaps</span> <span class="op">&lt;-</span> <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps</a></span><span class="op">(</span><span class="va">gs</span>, <span class="va">cnv.hmm</span><span class="op">)</span>
<span class="va">joined2</span> <span class="op">&lt;-</span> <span class="va">gs</span><span class="op">[</span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">queryHits</a></span><span class="op">(</span><span class="va">olaps</span><span class="op">)</span><span class="op">]</span>
<span class="va">joined2</span><span class="op">$</span><span class="va">score</span> <span class="op">&lt;-</span> <span class="va">cnv.hmm</span><span class="op">$</span><span class="va">score</span><span class="op">[</span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">subjectHits</a></span><span class="op">(</span><span class="va">olaps</span><span class="op">)</span><span class="op">]</span>
<span class="va">joined2</span>
<span class="co">#&gt; GRanges object with 22366 ranges and 2 metadata columns:</span>
<span class="co">#&gt;         seqnames              ranges strand |     gene_id     score</span>
<span class="co">#&gt;            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;       1    chr19   63549984-63565932      - |           1      3.18</span>
<span class="co">#&gt;      10     chr8   18293035-18303003      + |          10      1.96</span>
<span class="co">#&gt;     100    chr20   42681577-42713790      - |         100      2.98</span>
<span class="co">#&gt;    1000    chr18   23784933-24011189      - |        1000      2.94</span>
<span class="co">#&gt;   10000     chr1 241718158-242073207      - |       10000      5.92</span>
<span class="co">#&gt;     ...      ...                 ...    ... .         ...       ...</span>
<span class="co">#&gt;    9990    chr15   32309489-32417557      - |        9990      2.26</span>
<span class="co">#&gt;    9991     chr9 114020538-114135733      - |        9991      3.45</span>
<span class="co">#&gt;    9993    chr22   17403795-17489967      - |        9993      4.48</span>
<span class="co">#&gt;    9994     chr6   90596334-90640876      + |        9994      3.14</span>
<span class="co">#&gt;    9997    chr22   49308863-49310900      - |        9997      3.35</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 49 sequences (1 circular) from hg18 genome</span></code></pre></div>
<p>We then restrict both tracks to common genes.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">isect</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">joined</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">joined2</span><span class="op">)</span><span class="op">)</span>
<span class="va">joined</span> <span class="op">&lt;-</span> <span class="va">joined</span><span class="op">[</span><span class="va">isect</span><span class="op">]</span>
<span class="va">joined2</span> <span class="op">&lt;-</span> <span class="va">joined2</span><span class="op">[</span><span class="va">isect</span><span class="op">]</span></code></pre></div>
<p>Now, can assess agreement by testing the correlation between
SNP-inferred gene copy numbers and the corresponding sequencing-inferred
gene copy numbers.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">joined</span><span class="op">$</span><span class="va">score</span>, <span class="va">joined2</span><span class="op">$</span><span class="va">score</span><span class="op">)</span>
<span class="co">#&gt; [1] 0.7953383</span>
<span class="fu"><a href="https://rdrr.io/r/stats/cor.test.html" class="external-link">cor.test</a></span><span class="op">(</span><span class="va">joined</span><span class="op">$</span><span class="va">score</span>, <span class="va">joined2</span><span class="op">$</span><span class="va">score</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  Pearson's product-moment correlation</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; data:  joined$score and joined2$score</span>
<span class="co">#&gt; t = 176.07, df = 18008, p-value &lt; 2.2e-16</span>
<span class="co">#&gt; alternative hypothesis: true correlation is not equal to 0</span>
<span class="co">#&gt; 95 percent confidence interval:</span>
<span class="co">#&gt;  0.7899089 0.8006431</span>
<span class="co">#&gt; sample estimates:</span>
<span class="co">#&gt;       cor </span>
<span class="co">#&gt; 0.7953383</span></code></pre></div>
<p>We also inspect the correlation via a boxplot.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">spl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span><span class="va">joined2</span><span class="op">$</span><span class="va">score</span>, <span class="va">joined</span><span class="op">$</span><span class="va">score</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/boxplot.html" class="external-link">boxplot</a></span><span class="op">(</span><span class="va">spl</span>, xlab <span class="op">=</span> <span class="st">"SNP-inferred copy number"</span>, ylab <span class="op">=</span> <span class="st">"sequencing-inferred copy number"</span><span class="op">)</span>
<span class="va">rho</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">joined</span><span class="op">$</span><span class="va">score</span>, <span class="va">joined2</span><span class="op">$</span><span class="va">score</span><span class="op">)</span>
<span class="va">rho</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"cor"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">rho</span>, digits<span class="op">=</span><span class="fl">3</span><span class="op">)</span>, sep<span class="op">=</span><span class="st">" = "</span><span class="op">)</span>

<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.test.html" class="external-link">cor.test</a></span><span class="op">(</span><span class="va">joined</span><span class="op">$</span><span class="va">score</span>, <span class="va">joined2</span><span class="op">$</span><span class="va">score</span><span class="op">)</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p</span><span class="op">$</span><span class="va">p.value</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="st">"   p &lt; 2.2e-16"</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span><span class="st">"topright"</span>, legend<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">rho</span>, <span class="va">p</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="BasicChecks_files/figure-html/unnamed-chunk-11-1.png" width="768"></p>
</div>
<div class="section level2">
<h2 id="check-expressed-genes-are-showing-up-as-prey-293t-cells">Check: expressed genes are showing up as prey (293T cells)<a class="anchor" aria-label="anchor" href="#check-expressed-genes-are-showing-up-as-prey-293t-cells"></a>
</h2>
<p>Get RNA-seq data for HEK293 cells from GEO: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122425" class="external-link">GSE122425</a></p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getGSE122425.html">getGSE122425</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Using cached version from 2023-02-20 10:51:22</span>
<span class="va">se</span>
<span class="co">#&gt; class: SummarizedExperiment </span>
<span class="co">#&gt; dim: 57905 6 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(2): raw rpkm</span>
<span class="co">#&gt; rownames(57905): ENSG00000223972 ENSG00000227232 ... ENSG00000231514</span>
<span class="co">#&gt;   ENSG00000235857</span>
<span class="co">#&gt; rowData names(4): SYMBOL KO GO length</span>
<span class="co">#&gt; colnames(6): GSM3466389 GSM3466390 ... GSM3466393 GSM3466394</span>
<span class="co">#&gt; colData names(41): title geo_accession ... passages.ch1 strain.ch1</span></code></pre></div>
<p>Inspect expression of prey genes:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bait</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">bp.293t</span><span class="op">$</span><span class="va">SymbolA</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">bait</span><span class="op">)</span>
<span class="co">#&gt; [1] 8995</span>
<span class="va">prey</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">bp.293t</span><span class="op">$</span><span class="va">SymbolB</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">prey</span><span class="op">)</span>
<span class="co">#&gt; [1] 10419</span></code></pre></div>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">prey</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">$</span><span class="va">SYMBOL</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>las <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/boxplot.html" class="external-link">boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"rpkm"</span><span class="op">)</span> <span class="op">+</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">[</span><span class="va">ind</span>,<span class="op">]</span>, 
        names <span class="op">=</span> <span class="va">se</span><span class="op">$</span><span class="va">title</span>, 
        ylab <span class="op">=</span> <span class="st">"log2 RPKM"</span><span class="op">)</span></code></pre></div>
<p><img src="BasicChecks_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>How many prey genes are expressed (raw read count &gt; 0) in all 3 WT
reps:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># background: how many genes in total are expressed in all three WT reps</span>
<span class="va">gr0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">gr0</span> <span class="op">==</span> <span class="fl">3</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; FALSE  TRUE </span>
<span class="co">#&gt; 33842 24063</span>
<span class="co"># prey: expressed in all three WT reps</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">gr0</span><span class="op">[</span><span class="va">ind</span><span class="op">]</span> <span class="op">==</span> <span class="fl">3</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; FALSE  TRUE </span>
<span class="co">#&gt;   599  9346</span>
<span class="co"># prey: expressed in at least one WT rep</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">gr0</span><span class="op">[</span><span class="va">ind</span><span class="op">]</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; FALSE  TRUE </span>
<span class="co">#&gt;   305  9640</span></code></pre></div>
<p>Are prey genes overrepresented in the expressed genes?</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exprTable</span> <span class="op">&lt;-</span>
     <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">9346</span>, <span class="fl">1076</span>, <span class="fl">14717</span>, <span class="fl">32766</span><span class="op">)</span>,
            nrow <span class="op">=</span> <span class="fl">2</span>,
            dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Expressed"</span>, <span class="st">"Not.expressed"</span><span class="op">)</span>,
                            <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"In.prey.set"</span>, <span class="st">"Not.in.prey.set"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">exprTable</span>
<span class="co">#&gt;               In.prey.set Not.in.prey.set</span>
<span class="co">#&gt; Expressed            9346           14717</span>
<span class="co">#&gt; Not.expressed        1076           32766</span></code></pre></div>
<p>Test using hypergeometric test (i.e. one-sided Fisher’s exact
test):</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/fisher.test.html" class="external-link">fisher.test</a></span><span class="op">(</span><span class="va">exprTable</span>, alternative <span class="op">=</span> <span class="st">"greater"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  Fisher's Exact Test for Count Data</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; data:  exprTable</span>
<span class="co">#&gt; p-value &lt; 2.2e-16</span>
<span class="co">#&gt; alternative hypothesis: true odds ratio is greater than 1</span>
<span class="co">#&gt; 95 percent confidence interval:</span>
<span class="co">#&gt;  18.29105      Inf</span>
<span class="co">#&gt; sample estimates:</span>
<span class="co">#&gt; odds ratio </span>
<span class="co">#&gt;   19.34726</span></code></pre></div>
<p>Alternatively: permutation test, i.e. repeatedly sample number of
prey genes from the background, and assess how often we have as many or
more than 9346 genes expressed:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">permgr0</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gr0</span>, <span class="va">nr.genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">prey</span><span class="op">)</span><span class="op">)</span> 
<span class="op">{</span>
    <span class="va">ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">gr0</span><span class="op">)</span>, <span class="va">nr.genes</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">gr0</span><span class="op">[</span><span class="va">ind</span><span class="op">]</span> <span class="op">==</span> <span class="fl">3</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">perms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">replicate</a></span><span class="op">(</span><span class="fu">permgr0</span><span class="op">(</span><span class="va">gr0</span><span class="op">)</span>, <span class="fl">1000</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">perms</span><span class="op">)</span>
<span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span>
<span class="co">#&gt;    1000    1000    1000    1000    1000    1000</span>
<span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">perms</span> <span class="op">&gt;=</span> <span class="fl">9346</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span> <span class="op">/</span> <span class="fl">1001</span>
<span class="co">#&gt; [1] 0.000999001</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="check-is-there-a-relationship-between-prey-frequency-and-prey-expression-level">Check: is there a relationship between prey frequency and prey
expression level?<a class="anchor" aria-label="anchor" href="#check-is-there-a-relationship-between-prey-frequency-and-prey-expression-level"></a>
</h2>
<p>Check which genes turn up most frequently as prey:</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">prey.freq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">bp.293t</span><span class="op">$</span><span class="va">SymbolB</span><span class="op">)</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">preys</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">prey.freq</span><span class="op">)</span>
<span class="va">prey.freq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">prey.freq</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">prey.freq</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">preys</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">prey.freq</span><span class="op">)</span>
<span class="co">#&gt; HSPA5 HSPA8 TUBB8   UBB  YBX1 YWHAH </span>
<span class="co">#&gt;   199   192   176   173   139   132</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">prey.freq</span><span class="op">)</span>
<span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span>
<span class="co">#&gt;    1.00    2.00    6.00   11.34   16.00  199.00</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">prey.freq</span>, breaks <span class="op">=</span> <span class="fl">50</span>, main <span class="op">=</span> <span class="st">""</span>, xlab <span class="op">=</span> <span class="st">"Number of PPIs"</span>, ylab <span class="op">=</span> <span class="st">"Number of genes"</span><span class="op">)</span></code></pre></div>
<p><img src="BasicChecks_files/figure-html/prey-freq-1.png" width="700"></p>
<p>Prey genes are involved in 11 PPIs on average.</p>
<p>There doesn’t seem to be a strong correlation between expression
level and the frequency of gene to turn up as prey:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">prey.freq</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">$</span><span class="va">SYMBOL</span><span class="op">)</span>
<span class="va">rmeans</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"rpkm"</span><span class="op">)</span><span class="op">[</span><span class="va">ind</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span>
<span class="va">log.rmeans</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">rmeans</span> <span class="op">+</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>pch <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span> x <span class="op">=</span> <span class="va">prey.freq</span>,
      y <span class="op">=</span> <span class="va">log.rmeans</span>,
      xlab <span class="op">=</span> <span class="st">"prey frequency"</span>,
      ylab <span class="op">=</span> <span class="st">"log2 RPKM"</span><span class="op">)</span></code></pre></div>
<p><img src="BasicChecks_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">prey.freq</span>, 
    <span class="va">log.rmeans</span>,
    use <span class="op">=</span> <span class="st">"pairwise.complete.obs"</span><span class="op">)</span>
<span class="co">#&gt; [1] 0.2035977</span></code></pre></div>
<p>See also the <a href="https://ccb-hms.github.io/BioPlexAnalysis/articles/BioNet.html">BioNet
maximum scoring subnetwork analysis vignette</a> for a more
comprehensive analysis of the 293T transcriptome data from GSE122425
when mapped onto BioPlex PPI network.</p>
</div>
<div class="section level2">
<h2 id="check-differential-protein-expression-hek293-vs--hct116">Check: differential protein expression (HEK293 vs. HCT116)<a class="anchor" aria-label="anchor" href="#check-differential-protein-expression-hek293-vs--hct116"></a>
</h2>
<p>Get the relative protein expression data comparing 293T and HCT116
cells from Supplementary Table S4A of the BioPlex 3 paper:</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.prot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getBioplexProteome.html">getBioplexProteome</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Using cached version from 2023-02-20 10:51:29</span>
<span class="va">bp.prot</span>
<span class="co">#&gt; class: SummarizedExperiment </span>
<span class="co">#&gt; dim: 9604 10 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): exprs</span>
<span class="co">#&gt; rownames(9604): P0CG40 Q8IXZ3-4 ... Q9H3S5 Q8WYQ3</span>
<span class="co">#&gt; rowData names(5): ENTREZID SYMBOL nr.peptides log2ratio adj.pvalue</span>
<span class="co">#&gt; colnames(10): HCT1 HCT2 ... HEK4 HEK5</span>
<span class="co">#&gt; colData names(1): cell.line</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 9604 rows and 5 columns</span>
<span class="co">#&gt;             ENTREZID      SYMBOL nr.peptides log2ratio  adj.pvalue</span>
<span class="co">#&gt;          &lt;character&gt; &lt;character&gt;   &lt;integer&gt; &lt;numeric&gt;   &lt;numeric&gt;</span>
<span class="co">#&gt; P0CG40     100131390         SP9           1 -2.819071 6.66209e-08</span>
<span class="co">#&gt; Q8IXZ3-4      221833         SP8           3 -3.419888 6.94973e-07</span>
<span class="co">#&gt; P55011          6558     SLC12A2           4  0.612380 4.85602e-06</span>
<span class="co">#&gt; O60341         23028       KDM1A           7 -0.319695 5.08667e-04</span>
<span class="co">#&gt; O14654          8471        IRS4           4 -5.951096 1.45902e-06</span>
<span class="co">#&gt; ...              ...         ...         ...       ...         ...</span>
<span class="co">#&gt; Q9H6X4         80194     TMEM134           2 -0.379342 7.67195e-05</span>
<span class="co">#&gt; Q9BS91         55032     SLC35A5           1 -2.237634 8.75523e-05</span>
<span class="co">#&gt; Q9UKJ5         26511       CHIC2           1 -0.614932 1.78756e-03</span>
<span class="co">#&gt; Q9H3S5         93183        PIGM           1 -1.011397 8.91589e-06</span>
<span class="co">#&gt; Q8WYQ3        400916     CHCHD10           1  0.743852 1.17163e-03</span></code></pre></div>
<p>A couple of quick sanity checks:</p>
<ol style="list-style-type: decimal">
<li>The relative abundances are scaled to sum up to 100% for each
protein:</li>
</ol>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span><span class="op">)</span> 
<span class="co">#&gt;    P0CG40  Q8IXZ3-4    P55011    O60341    O14654 </span>
<span class="co">#&gt;  99.99994  99.99991  99.99996 100.00011 100.00006</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>The <code>rowData</code> column <code>log2ratio</code> corresponds
to the mean of the five HEK samples, divided by the mean of the five HCT
samples (and then taking log2 of it):</li>
</ol>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ratio</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">6</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">ratio</span><span class="op">)</span>
<span class="co">#&gt;     P0CG40   Q8IXZ3-4     P55011     O60341     O14654 </span>
<span class="co">#&gt; -2.8190710 -3.4198879  0.6123799 -0.3196953 -5.9510960</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>The <code>rowData</code> column <code>adj.pvalue</code> stores
Benjamini-Hochberg adjusted <em>p</em>-values from a <em>t</em>-test
between the five HEK samples and the five HCT samples:</li>
</ol>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/t.test.html" class="external-link">t.test</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span>, <span class="fl">6</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  Welch Two Sample t-test</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; data:  assay(bp.prot)[1, 1:5] and assay(bp.prot)[1, 6:10]</span>
<span class="co">#&gt; t = -27.898, df = 7.5779, p-value = 6.482e-09</span>
<span class="co">#&gt; alternative hypothesis: true difference in means is not equal to 0</span>
<span class="co">#&gt; 95 percent confidence interval:</span>
<span class="co">#&gt;  -16.29035 -13.78047</span>
<span class="co">#&gt; sample estimates:</span>
<span class="co">#&gt; mean of x mean of y </span>
<span class="co">#&gt;  2.482288 17.517700</span></code></pre></div>
<p>The <a href="https://ccb-hms.github.io/BioPlexAnalysis/articles/TranscriptomeProteome.html">Transcriptome-Proteome
analysis vignette</a> also explores the agreement between differential
gene expression and differential protein expression when comparing
HEK293 against HCT116 cells.</p>
</div>
<div class="section level2">
<h2 id="sessioninfo">SessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.2.0 (2022-04-22)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04.5 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span>
<span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">#&gt; [8] base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] GenomicFeatures_1.50.4      graph_1.76.0               </span>
<span class="co">#&gt;  [3] AnnotationHub_3.6.0         BiocFileCache_2.6.1        </span>
<span class="co">#&gt;  [5] dbplyr_2.2.0                AnnotationDbi_1.60.0       </span>
<span class="co">#&gt;  [7] BioPlex_1.5.4               SummarizedExperiment_1.28.0</span>
<span class="co">#&gt;  [9] Biobase_2.58.0              GenomicRanges_1.50.2       </span>
<span class="co">#&gt; [11] GenomeInfoDb_1.34.9         IRanges_2.32.0             </span>
<span class="co">#&gt; [13] S4Vectors_0.36.1            BiocGenerics_0.44.0        </span>
<span class="co">#&gt; [15] MatrixGenerics_1.10.0       matrixStats_0.62.0         </span>
<span class="co">#&gt; [17] BiocStyle_2.26.0           </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] bitops_1.0-7                  fs_1.5.2                     </span>
<span class="co">#&gt;  [3] bit64_4.0.5                   progress_1.2.2               </span>
<span class="co">#&gt;  [5] filelock_1.0.2                httr_1.4.3                   </span>
<span class="co">#&gt;  [7] rprojroot_2.0.3               tools_4.2.0                  </span>
<span class="co">#&gt;  [9] bslib_0.3.1                   utf8_1.2.2                   </span>
<span class="co">#&gt; [11] R6_2.5.1                      DBI_1.1.3                    </span>
<span class="co">#&gt; [13] prettyunits_1.1.1             tidyselect_1.1.2             </span>
<span class="co">#&gt; [15] bit_4.0.4                     curl_4.3.2                   </span>
<span class="co">#&gt; [17] compiler_4.2.0                textshaping_0.3.6            </span>
<span class="co">#&gt; [19] cli_3.3.0                     xml2_1.3.3                   </span>
<span class="co">#&gt; [21] desc_1.4.1                    DelayedArray_0.24.0          </span>
<span class="co">#&gt; [23] rtracklayer_1.58.0            bookdown_0.27                </span>
<span class="co">#&gt; [25] sass_0.4.1                    rappdirs_0.3.3               </span>
<span class="co">#&gt; [27] pkgdown_2.0.4                 Rsamtools_2.14.0             </span>
<span class="co">#&gt; [29] systemfonts_1.0.4             stringr_1.4.0                </span>
<span class="co">#&gt; [31] digest_0.6.29                 rmarkdown_2.14               </span>
<span class="co">#&gt; [33] XVector_0.38.0                pkgconfig_2.0.3              </span>
<span class="co">#&gt; [35] htmltools_0.5.2               highr_0.9                    </span>
<span class="co">#&gt; [37] fastmap_1.1.0                 rlang_1.0.2                  </span>
<span class="co">#&gt; [39] RSQLite_2.2.14                shiny_1.7.1                  </span>
<span class="co">#&gt; [41] BiocIO_1.8.0                  jquerylib_0.1.4              </span>
<span class="co">#&gt; [43] generics_0.1.2                jsonlite_1.8.0               </span>
<span class="co">#&gt; [45] BiocParallel_1.32.5           dplyr_1.0.9                  </span>
<span class="co">#&gt; [47] RCurl_1.98-1.7                magrittr_2.0.3               </span>
<span class="co">#&gt; [49] GenomeInfoDbData_1.2.9        Matrix_1.4-1                 </span>
<span class="co">#&gt; [51] Rcpp_1.0.8.3                  fansi_1.0.3                  </span>
<span class="co">#&gt; [53] lifecycle_1.0.1               stringi_1.7.6                </span>
<span class="co">#&gt; [55] yaml_2.3.5                    zlibbioc_1.44.0              </span>
<span class="co">#&gt; [57] grid_4.2.0                    blob_1.2.3                   </span>
<span class="co">#&gt; [59] parallel_4.2.0                promises_1.2.0.1             </span>
<span class="co">#&gt; [61] ExperimentHub_2.6.0           crayon_1.5.1                 </span>
<span class="co">#&gt; [63] lattice_0.20-45               Biostrings_2.66.0            </span>
<span class="co">#&gt; [65] hms_1.1.1                     KEGGREST_1.38.0              </span>
<span class="co">#&gt; [67] knitr_1.39                    pillar_1.7.0                 </span>
<span class="co">#&gt; [69] rjson_0.2.21                  codetools_0.2-18             </span>
<span class="co">#&gt; [71] biomaRt_2.54.0                XML_3.99-0.10                </span>
<span class="co">#&gt; [73] glue_1.6.2                    BiocVersion_3.16.0           </span>
<span class="co">#&gt; [75] evaluate_0.15                 BiocManager_1.30.18          </span>
<span class="co">#&gt; [77] png_0.1-7                     vctrs_0.4.1                  </span>
<span class="co">#&gt; [79] httpuv_1.6.5                  purrr_0.3.4                  </span>
<span class="co">#&gt; [81] assertthat_0.2.1              cachem_1.0.6                 </span>
<span class="co">#&gt; [83] xfun_0.31                     mime_0.12                    </span>
<span class="co">#&gt; [85] xtable_1.8-4                  restfulr_0.0.15              </span>
<span class="co">#&gt; [87] later_1.3.0                   ragg_1.2.2                   </span>
<span class="co">#&gt; [89] tibble_3.1.7                  GenomicAlignments_1.34.0     </span>
<span class="co">#&gt; [91] memoise_2.0.1                 ellipsis_0.3.2               </span>
<span class="co">#&gt; [93] interactiveDisplayBase_1.36.0</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Ludwig Geistlinger, Robert Gentleman.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.4.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
