<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Import and representation of BioPlex PPI data and related resources • BioPlex</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Import and representation of BioPlex PPI data and related resources">
<meta property="og:description" content="BioPlex">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BioPlex</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.5.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/BioPlex.html">Data retrieval</a>
</li>
<li>
  <a href="../articles/BasicChecks.html">Data checks</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ccb-hms/BioPlex" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Import and representation of BioPlex PPI data
and related resources</h1>
                        <h4 data-toc-skip class="author">Ludwig
Geistlinger and Robert Gentleman</h4>
            
            <h4 data-toc-skip class="date">26 February 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ccb-hms/BioPlex/blob/HEAD/vignettes/BioPlex.Rmd" class="external-link"><code>vignettes/BioPlex.Rmd</code></a></small>
      <div class="hidden name"><code>BioPlex.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<p>The <a href="https://bioplex.hms.harvard.edu/" class="external-link">BioPlex project</a>
uses affinity-purification mass spectrometry to profile protein-protein
interactions (PPIs) in human cell lines.</p>
<p>To date, the BioPlex project has created two proteome-scale,
cell-line-specific PPI networks. The first, BioPlex 3.0, results from
affinity purification of 10,128 human proteins —- half the proteome —-
in 293T cells and includes 118,162 interactions among 14,586 proteins.
The second results from 5,522 immunoprecipitations in HCT116 cells and
includes 70,966 interactions between 10,531 proteins.</p>
<p>For more information, please see:</p>
<ul>
<li>Huttlin et al. <a href="https://doi.org/10.1016/j.cell.2015.06.043" class="external-link">The BioPlex network: a
systematic exploration of the human interactome</a>. <em>Cell</em>,
2015.</li>
<li>Huttlin et al. <a href="https://doi.org/10.1038/nature22366" class="external-link">Architecture of the human
interactome defines protein communities and disease networks</a>,
<em>Nature</em>, 2017.</li>
<li>Huttlin et al. <a href="https://doi.org/10.1016/j.cell.2021.04.011" class="external-link">Dual proteome-scale
networks reveal cell-specific remodeling of the human interactome</a>,
<em>Cell</em>, 2021.</li>
</ul>
<p>The <a href="https://bioconductor.org/packages/BioPlex" class="external-link">BioPlex R
package</a> implements access to the BioPlex protein-protein interaction
networks and related resources from within R. Besides protein-protein
interaction networks for 293T and HCT116 cells, this includes access to
<a href="http://mips.helmholtz-muenchen.de/corum" class="external-link">CORUM</a> protein
complex data, and transcriptome and proteome data for the two cell
lines.</p>
<p>Functionality focuses on importing these data resources and storing
them in dedicated Bioconductor data structures, as a foundation for
integrative downstream analysis of the data. For a set of downstream
analyses and applications, please see the <a href="https://github.com/ccb-hms/BioPlexAnalysis" class="external-link">BioPlexAnalysis
package</a> and <a href="https://ccb-hms.github.io/BioPlexAnalysis/">analysis
vignettes</a>.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install the package, start R and enter:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"BioPlex"</span><span class="op">)</span></code></pre></div>
<p>After the installation, we proceed by loading the package and
additional packages used in the vignette.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ccb-hms/BioPlex" class="external-link">BioPlex</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ExperimentHub</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">graph</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-resources">Data resources<a class="anchor" aria-label="anchor" href="#data-resources"></a>
</h2>
<div class="section level3">
<h3 id="general">General<a class="anchor" aria-label="anchor" href="#general"></a>
</h3>
<p>Connect to <a href="http://bioconductor.org/packages/AnnotationHub" class="external-link">AnnotationHub</a>:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ah</span> <span class="op">&lt;-</span> <span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>Connect to <a href="http://bioconductor.org/packages/ExperimentHub" class="external-link">ExperimentHub</a>:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu">ExperimentHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>OrgDb package for human:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orgdb</span> <span class="op">&lt;-</span> <span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"orgDb"</span>, <span class="st">"Homo sapiens"</span><span class="op">)</span><span class="op">)</span>
<span class="va">orgdb</span> <span class="op">&lt;-</span> <span class="va">orgdb</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="va">orgdb</span>
<span class="co">#&gt; OrgDb object:</span>
<span class="co">#&gt; | DBSCHEMAVERSION: 2.1</span>
<span class="co">#&gt; | Db type: OrgDb</span>
<span class="co">#&gt; | Supporting package: AnnotationDbi</span>
<span class="co">#&gt; | DBSCHEMA: HUMAN_DB</span>
<span class="co">#&gt; | ORGANISM: Homo sapiens</span>
<span class="co">#&gt; | SPECIES: Human</span>
<span class="co">#&gt; | EGSOURCEDATE: 2022-Sep12</span>
<span class="co">#&gt; | EGSOURCENAME: Entrez Gene</span>
<span class="co">#&gt; | EGSOURCEURL: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA</span>
<span class="co">#&gt; | CENTRALID: EG</span>
<span class="co">#&gt; | TAXID: 9606</span>
<span class="co">#&gt; | GOSOURCENAME: Gene Ontology</span>
<span class="co">#&gt; | GOSOURCEURL: http://current.geneontology.org/ontology/go-basic.obo</span>
<span class="co">#&gt; | GOSOURCEDATE: 2022-07-01</span>
<span class="co">#&gt; | GOEGSOURCEDATE: 2022-Sep12</span>
<span class="co">#&gt; | GOEGSOURCENAME: Entrez Gene</span>
<span class="co">#&gt; | GOEGSOURCEURL: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA</span>
<span class="co">#&gt; | KEGGSOURCENAME: KEGG GENOME</span>
<span class="co">#&gt; | KEGGSOURCEURL: ftp://ftp.genome.jp/pub/kegg/genomes</span>
<span class="co">#&gt; | KEGGSOURCEDATE: 2011-Mar15</span>
<span class="co">#&gt; | GPSOURCENAME: UCSC Genome Bioinformatics (Homo sapiens)</span>
<span class="co">#&gt; | GPSOURCEURL: </span>
<span class="co">#&gt; | GPSOURCEDATE: 2022-Aug31</span>
<span class="co">#&gt; | ENSOURCEDATE: 2022-Jun28</span>
<span class="co">#&gt; | ENSOURCENAME: Ensembl</span>
<span class="co">#&gt; | ENSOURCEURL: ftp://ftp.ensembl.org/pub/current_fasta</span>
<span class="co">#&gt; | UPSOURCENAME: Uniprot</span>
<span class="co">#&gt; | UPSOURCEURL: http://www.UniProt.org/</span>
<span class="co">#&gt; | UPSOURCEDATE: Fri Sep 23 16:26:35 2022</span>
<span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">keytypes</a></span><span class="op">(</span><span class="va">orgdb</span><span class="op">)</span>
<span class="co">#&gt;  [1] "ACCNUM"       "ALIAS"        "ENSEMBL"      "ENSEMBLPROT"  "ENSEMBLTRANS"</span>
<span class="co">#&gt;  [6] "ENTREZID"     "ENZYME"       "EVIDENCE"     "EVIDENCEALL"  "GENENAME"    </span>
<span class="co">#&gt; [11] "GENETYPE"     "GO"           "GOALL"        "IPI"          "MAP"         </span>
<span class="co">#&gt; [16] "OMIM"         "ONTOLOGY"     "ONTOLOGYALL"  "PATH"         "PFAM"        </span>
<span class="co">#&gt; [21] "PMID"         "PROSITE"      "REFSEQ"       "SYMBOL"       "UCSCKG"      </span>
<span class="co">#&gt; [26] "UNIPROT"</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="bioplex-ppis">BioPlex PPIs<a class="anchor" aria-label="anchor" href="#bioplex-ppis"></a>
</h3>
<p><a href="https://bioplex.hms.harvard.edu/interactions.php" class="external-link">Available
networks</a> include:</p>
<ul>
<li>BioPlex PPI network for human embryonic kidney <a href="https://en.wikipedia.org/wiki/293T" class="external-link">293T</a> cells (versions 1.0,
2.0, and 3.0)</li>
<li>BioPlex PPI network for human colon cancer <a href="https://en.wikipedia.org/wiki/HCT116_cells" class="external-link">HCT116</a> cells
(version 1.0)</li>
</ul>
<p>Let’s get the latest version of the 293T PPI network:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.293t</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getBioPlex.html">getBioPlex</a></span><span class="op">(</span>cell.line <span class="op">=</span> <span class="st">"293T"</span>, version <span class="op">=</span> <span class="st">"3.0"</span><span class="op">)</span>
<span class="co">#&gt; Using cached version from 2023-02-26 15:10:40</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">bp.293t</span><span class="op">)</span>
<span class="co">#&gt;    GeneA  GeneB UniprotA UniprotB SymbolA SymbolB           pW          pNI</span>
<span class="co">#&gt; 1    100 728378   P00813   A5A3E0     ADA   POTEF 6.881844e-10 0.0001176357</span>
<span class="co">#&gt; 2 222389   6137 Q8N7W2-2   P26373   BEND7   RPL13 1.340380e-18 0.2256644741</span>
<span class="co">#&gt; 3 222389   5928 Q8N7W2-2 Q09028-3   BEND7   RBBP4 7.221401e-21 0.0000641669</span>
<span class="co">#&gt; 4 222389  25873 Q8N7W2-2   Q9Y3U8   BEND7   RPL36 7.058372e-17 0.1281827343</span>
<span class="co">#&gt; 5 222389   6124 Q8N7W2-2   P36578   BEND7    RPL4 1.632313e-22 0.2006379109</span>
<span class="co">#&gt; 6 222389   6188 Q8N7W2-2   P23396   BEND7    RPS3 3.986270e-26 0.0010264311</span>
<span class="co">#&gt;        pInt</span>
<span class="co">#&gt; 1 0.9998824</span>
<span class="co">#&gt; 2 0.7743355</span>
<span class="co">#&gt; 3 0.9999358</span>
<span class="co">#&gt; 4 0.8718173</span>
<span class="co">#&gt; 5 0.7993621</span>
<span class="co">#&gt; 6 0.9989736</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">bp.293t</span><span class="op">)</span>
<span class="co">#&gt; [1] 118162</span></code></pre></div>
<p>Each row corresponds to a PPI between a bait protein A and a prey
protein B, for which NCBI Entrez Gene IDs, Uniprot IDs, and gene symbols
are annotated. The last three columns reflect the likelihood that each
interaction resulted from either an incorrect protein identification
(<code>pW</code>), background (<code>pNI</code>), or a bona fide
interacting partner (<code>pInt</code>) as determined using the <a href="https://github.com/dnusinow/cRomppass" class="external-link">CompPASS algorithm</a>.</p>
<p>Analgously, we can obtain the latest version of the HCT116 PPI
network:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.hct116</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getBioPlex.html">getBioPlex</a></span><span class="op">(</span>cell.line <span class="op">=</span> <span class="st">"HCT116"</span>, version <span class="op">=</span> <span class="st">"1.0"</span><span class="op">)</span>
<span class="co">#&gt; Using cached version from 2023-02-26 15:11:17</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">bp.hct116</span><span class="op">)</span>
<span class="co">#&gt;   GeneA  GeneB UniprotA UniprotB  SymbolA SymbolB           pW          pNI</span>
<span class="co">#&gt; 1 88455  50649   Q8IZ07 Q9NR80-4 ANKRD13A ARHGEF4 3.959215e-04 3.298003e-05</span>
<span class="co">#&gt; 2 88455 115106   Q8IZ07   Q96CS2 ANKRD13A   HAUS1 4.488473e-02 1.934731e-03</span>
<span class="co">#&gt; 3 88455  23086   Q8IZ07 Q8NEV8-2 ANKRD13A   EXPH5 7.402394e-05 9.296226e-04</span>
<span class="co">#&gt; 4 88455  54930   Q8IZ07   Q9H6D7 ANKRD13A   HAUS4 9.180959e-07 1.278318e-04</span>
<span class="co">#&gt; 5 88455  79441   Q8IZ07   Q68CZ6 ANKRD13A   HAUS3 8.709394e-07 1.495480e-03</span>
<span class="co">#&gt; 6 88455  93323   Q8IZ07 Q9BT25-2 ANKRD13A   HAUS8 9.147659e-06 2.061483e-03</span>
<span class="co">#&gt;        pInt</span>
<span class="co">#&gt; 1 0.9995711</span>
<span class="co">#&gt; 2 0.9531805</span>
<span class="co">#&gt; 3 0.9989964</span>
<span class="co">#&gt; 4 0.9998713</span>
<span class="co">#&gt; 5 0.9985036</span>
<span class="co">#&gt; 6 0.9979294</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">bp.hct116</span><span class="op">)</span>
<span class="co">#&gt; [1] 70966</span></code></pre></div>
<div class="section level4">
<h4 id="id-mapping">ID mapping<a class="anchor" aria-label="anchor" href="#id-mapping"></a>
</h4>
<p>The protein-to-gene mappings from BioPlex (i.e. UNIPROT-to-SYMBOL and
UNIPROT-to-ENTREZID) are based on the mappings available from Uniprot at
the time of publication of the BioPlex 3.0 networks.</p>
<p>We can update those based on Bioc annotation functionality:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.293t.remapped</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getBioPlex.html">getBioPlex</a></span><span class="op">(</span>cell.line <span class="op">=</span> <span class="st">"293T"</span>,
                               version <span class="op">=</span> <span class="st">"3.0"</span>,
                               remap.uniprot.ids <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; Using cached version from 2023-02-26 15:10:40</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="data-structures-for-bioplex-ppis">Data structures for BioPlex PPIs<a class="anchor" aria-label="anchor" href="#data-structures-for-bioplex-ppis"></a>
</h4>
<p>We can also represent a given version of the BioPlex PPI network for
a given cell line as one big graph where bait and prey relationship are
represented by directed edges from bait to prey.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bioplex2graph.html">bioplex2graph</a></span><span class="op">(</span><span class="va">bp.293t</span><span class="op">)</span>
<span class="va">bp.gr</span>
<span class="co">#&gt; A graphNEL graph with directed edges</span>
<span class="co">#&gt; Number of Nodes = 13689 </span>
<span class="co">#&gt; Number of Edges = 115868</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/nodeData-methods.html" class="external-link">nodeData</a></span><span class="op">(</span><span class="va">bp.gr</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; $P00813</span>
<span class="co">#&gt; $P00813$ENTREZID</span>
<span class="co">#&gt; [1] "100"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $P00813$SYMBOL</span>
<span class="co">#&gt; [1] "ADA"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $P00813$ISOFORM</span>
<span class="co">#&gt; [1] "P00813"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Q8N7W2</span>
<span class="co">#&gt; $Q8N7W2$ENTREZID</span>
<span class="co">#&gt; [1] "222389"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Q8N7W2$SYMBOL</span>
<span class="co">#&gt; [1] "BEND7"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Q8N7W2$ISOFORM</span>
<span class="co">#&gt; [1] "Q8N7W2-2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Q6ZMN8</span>
<span class="co">#&gt; $Q6ZMN8$ENTREZID</span>
<span class="co">#&gt; [1] "645121"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Q6ZMN8$SYMBOL</span>
<span class="co">#&gt; [1] "CCNI2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Q6ZMN8$ISOFORM</span>
<span class="co">#&gt; [1] "Q6ZMN8"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $P20138</span>
<span class="co">#&gt; $P20138$ENTREZID</span>
<span class="co">#&gt; [1] "945"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $P20138$SYMBOL</span>
<span class="co">#&gt; [1] "CD33"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $P20138$ISOFORM</span>
<span class="co">#&gt; [1] "P20138"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $P55039</span>
<span class="co">#&gt; $P55039$ENTREZID</span>
<span class="co">#&gt; [1] "1819"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $P55039$SYMBOL</span>
<span class="co">#&gt; [1] "DRG2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $P55039$ISOFORM</span>
<span class="co">#&gt; [1] "P55039"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Q17R55</span>
<span class="co">#&gt; $Q17R55$ENTREZID</span>
<span class="co">#&gt; [1] "148109"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Q17R55$SYMBOL</span>
<span class="co">#&gt; [1] "FAM187B"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Q17R55$ISOFORM</span>
<span class="co">#&gt; [1] "Q17R55"</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/edgeData-methods.html" class="external-link">edgeData</a></span><span class="op">(</span><span class="va">bp.gr</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; $`P00813|A5A3E0`</span>
<span class="co">#&gt; $`P00813|A5A3E0`$weight</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`P00813|A5A3E0`$pW</span>
<span class="co">#&gt; [1] 6.881844e-10</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`P00813|A5A3E0`$pNI</span>
<span class="co">#&gt; [1] 0.0001176357</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`P00813|A5A3E0`$pInt</span>
<span class="co">#&gt; [1] 0.9998824</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|P26373`</span>
<span class="co">#&gt; $`Q8N7W2|P26373`$weight</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|P26373`$pW</span>
<span class="co">#&gt; [1] 1.34038e-18</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|P26373`$pNI</span>
<span class="co">#&gt; [1] 0.2256645</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|P26373`$pInt</span>
<span class="co">#&gt; [1] 0.7743355</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|Q09028`</span>
<span class="co">#&gt; $`Q8N7W2|Q09028`$weight</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|Q09028`$pW</span>
<span class="co">#&gt; [1] 7.221401e-21</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|Q09028`$pNI</span>
<span class="co">#&gt; [1] 6.41669e-05</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|Q09028`$pInt</span>
<span class="co">#&gt; [1] 0.9999358</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|Q9Y3U8`</span>
<span class="co">#&gt; $`Q8N7W2|Q9Y3U8`$weight</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|Q9Y3U8`$pW</span>
<span class="co">#&gt; [1] 7.058372e-17</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|Q9Y3U8`$pNI</span>
<span class="co">#&gt; [1] 0.1281827</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|Q9Y3U8`$pInt</span>
<span class="co">#&gt; [1] 0.8718173</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|P36578`</span>
<span class="co">#&gt; $`Q8N7W2|P36578`$weight</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|P36578`$pW</span>
<span class="co">#&gt; [1] 1.632313e-22</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|P36578`$pNI</span>
<span class="co">#&gt; [1] 0.2006379</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|P36578`$pInt</span>
<span class="co">#&gt; [1] 0.7993621</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|P23396`</span>
<span class="co">#&gt; $`Q8N7W2|P23396`$weight</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|P23396`$pW</span>
<span class="co">#&gt; [1] 3.98627e-26</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|P23396`$pNI</span>
<span class="co">#&gt; [1] 0.001026431</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Q8N7W2|P23396`$pInt</span>
<span class="co">#&gt; [1] 0.9989736</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="pfam-domains">PFAM domains<a class="anchor" aria-label="anchor" href="#pfam-domains"></a>
</h4>
<p>We can easily add <a href="http://pfam.xfam.org" class="external-link">PFAM</a> domain
annotations to the node metadata:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotatePFAM.html">annotatePFAM</a></span><span class="op">(</span><span class="va">bp.gr</span>, <span class="va">orgdb</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/nodeData-methods.html" class="external-link">nodeData</a></span><span class="op">(</span><span class="va">bp.gr</span>, <span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/graphNEL-class.html" class="external-link">nodes</a></span><span class="op">(</span><span class="va">bp.gr</span><span class="op">)</span>, <span class="st">"PFAM"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; $P00813</span>
<span class="co">#&gt; [1] "PF00962"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Q8N7W2</span>
<span class="co">#&gt; [1] "PF10523"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Q6ZMN8</span>
<span class="co">#&gt; [1] "PF00134"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $P20138</span>
<span class="co">#&gt; [1] "PF00047" "PF07686"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $P55039</span>
<span class="co">#&gt; [1] "PF02824" "PF16897" "PF01926"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Q17R55</span>
<span class="co">#&gt; [1] NA</span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="corum-complexes">CORUM complexes<a class="anchor" aria-label="anchor" href="#corum-complexes"></a>
</h3>
<p>Obtain the complete set of human protein complexes from <a href="http://mips.helmholtz-muenchen.de/corum/#download" class="external-link">CORUM</a>:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getCorum.html">getCorum</a></span><span class="op">(</span>set <span class="op">=</span> <span class="st">"all"</span>, organism <span class="op">=</span> <span class="st">"Human"</span><span class="op">)</span>
<span class="co">#&gt; Using cached version from 2023-02-26 15:11:32</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">all</span><span class="op">)</span>
<span class="co">#&gt; [1] 2916   20</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">all</span><span class="op">)</span>
<span class="co">#&gt;  [1] "ComplexID"                           "ComplexName"                        </span>
<span class="co">#&gt;  [3] "Organism"                            "Synonyms"                           </span>
<span class="co">#&gt;  [5] "Cell.line"                           "subunits.UniProt.IDs."              </span>
<span class="co">#&gt;  [7] "subunits.Entrez.IDs."                "Protein.complex.purification.method"</span>
<span class="co">#&gt;  [9] "GO.ID"                               "GO.description"                     </span>
<span class="co">#&gt; [11] "FunCat.ID"                           "FunCat.description"                 </span>
<span class="co">#&gt; [13] "subunits.Gene.name."                 "Subunits.comment"                   </span>
<span class="co">#&gt; [15] "PubMed.ID"                           "Complex.comment"                    </span>
<span class="co">#&gt; [17] "Disease.comment"                     "SWISSPROT.organism"                 </span>
<span class="co">#&gt; [19] "subunits.Gene.name.syn."             "subunits.Protein.name."</span>
<span class="va">all</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>
<span class="co">#&gt;   ComplexID                           ComplexName Organism</span>
<span class="co">#&gt; 1         1                    BCL6-HDAC4 complex    Human</span>
<span class="co">#&gt; 2         2                    BCL6-HDAC5 complex    Human</span>
<span class="co">#&gt; 3         3                    BCL6-HDAC7 complex    Human</span>
<span class="co">#&gt; 4         4 Multisubunit ACTR coactivator complex    Human</span>
<span class="co">#&gt; 6        10                   Condensin I complex    Human</span>
<span class="co">#&gt;                Synonyms Cell.line</span>
<span class="co">#&gt; 1                  None      None</span>
<span class="co">#&gt; 2                  None      None</span>
<span class="co">#&gt; 3                  None      None</span>
<span class="co">#&gt; 4                  None      None</span>
<span class="co">#&gt; 6 13S condensin complex      None</span></code></pre></div>
<p>Core set of complexes:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">core</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getCorum.html">getCorum</a></span><span class="op">(</span>set <span class="op">=</span> <span class="st">"core"</span>, organism <span class="op">=</span> <span class="st">"Human"</span><span class="op">)</span>
<span class="co">#&gt; Using cached version from 2023-02-26 15:10:34</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">core</span><span class="op">)</span>
<span class="co">#&gt; [1] 2417   20</span></code></pre></div>
<p>Complexes with splice variants:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">splice</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getCorum.html">getCorum</a></span><span class="op">(</span>set <span class="op">=</span> <span class="st">"splice"</span>, organism <span class="op">=</span> <span class="st">"Human"</span><span class="op">)</span>
<span class="co">#&gt; Using cached version from 2023-02-26 15:11:34</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">splice</span><span class="op">)</span>
<span class="co">#&gt; [1] 44 20</span></code></pre></div>
<div class="section level4">
<h4 id="id-mapping-1">ID mapping<a class="anchor" aria-label="anchor" href="#id-mapping-1"></a>
</h4>
<p>The protein-to-gene mappings from CORUM (i.e. UNIPROT-to-SYMBOL and
UNIPROT-to-ENTREZID) might not be fully up-to-date.</p>
<p>We can update those based on Bioc annotation functionality:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">core.remapped</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getCorum.html">getCorum</a></span><span class="op">(</span>set <span class="op">=</span> <span class="st">"core"</span>, 
                          organism <span class="op">=</span> <span class="st">"Human"</span>,
                          remap.uniprot.ids <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; Using cached version from 2023-02-26 15:10:34</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="data-structures-for-corum-complexes">Data structures for CORUM complexes<a class="anchor" aria-label="anchor" href="#data-structures-for-corum-complexes"></a>
</h4>
<p>We can represent the CORUM complexes as a list of character vectors.
The names of the list are the complex IDs/names, and each element of the
list is a vector of UniProt IDs for each complex.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">core.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/corum2list.html">corum2list</a></span><span class="op">(</span><span class="va">core</span>, subunit.id.type <span class="op">=</span> <span class="st">"UNIPROT"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">core.list</span><span class="op">)</span>
<span class="co">#&gt; $`CORUM1_BCL6-HDAC4_complex`</span>
<span class="co">#&gt; [1] "P41182" "P56524"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`CORUM2_BCL6-HDAC5_complex`</span>
<span class="co">#&gt; [1] "P41182" "Q9UQL6"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`CORUM3_BCL6-HDAC7_complex`</span>
<span class="co">#&gt; [1] "P41182" "Q8WUI4"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $CORUM4_Multisubunit_ACTR_coactivator_complex</span>
<span class="co">#&gt; [1] "Q09472" "Q92793" "Q92831" "Q9Y6Q9"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`CORUM11_BLOC-3_(biogenesis_of_lysosome-related_organelles_complex_3)`</span>
<span class="co">#&gt; [1] "Q92902" "Q9NQG7"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`CORUM12_BLOC-2_(biogenesis_of_lysosome-related_organelles_complex_2)`</span>
<span class="co">#&gt; [1] "Q86YV9" "Q969F9" "Q9UPZ3"</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">core.list</span><span class="op">)</span>
<span class="co">#&gt; [1] 2417</span></code></pre></div>
<p>We can also represent the CORUM complexes as a list of graph
instances, where all nodes of a complex are connected to all other nodes
of that complex with undirected edges.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">core.glist</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/corum2graphlist.html">corum2graphlist</a></span><span class="op">(</span><span class="va">core</span>, subunit.id.type <span class="op">=</span> <span class="st">"UNIPROT"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">core.glist</span><span class="op">)</span>
<span class="co">#&gt; $`CORUM1_BCL6-HDAC4_complex`</span>
<span class="co">#&gt; A graphNEL graph with undirected edges</span>
<span class="co">#&gt; Number of Nodes = 2 </span>
<span class="co">#&gt; Number of Edges = 1 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`CORUM2_BCL6-HDAC5_complex`</span>
<span class="co">#&gt; A graphNEL graph with undirected edges</span>
<span class="co">#&gt; Number of Nodes = 2 </span>
<span class="co">#&gt; Number of Edges = 1 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`CORUM3_BCL6-HDAC7_complex`</span>
<span class="co">#&gt; A graphNEL graph with undirected edges</span>
<span class="co">#&gt; Number of Nodes = 2 </span>
<span class="co">#&gt; Number of Edges = 1 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $CORUM4_Multisubunit_ACTR_coactivator_complex</span>
<span class="co">#&gt; A graphNEL graph with undirected edges</span>
<span class="co">#&gt; Number of Nodes = 4 </span>
<span class="co">#&gt; Number of Edges = 6 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`CORUM11_BLOC-3_(biogenesis_of_lysosome-related_organelles_complex_3)`</span>
<span class="co">#&gt; A graphNEL graph with undirected edges</span>
<span class="co">#&gt; Number of Nodes = 2 </span>
<span class="co">#&gt; Number of Edges = 1 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`CORUM12_BLOC-2_(biogenesis_of_lysosome-related_organelles_complex_2)`</span>
<span class="co">#&gt; A graphNEL graph with undirected edges</span>
<span class="co">#&gt; Number of Nodes = 3 </span>
<span class="co">#&gt; Number of Edges = 3</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">core.glist</span><span class="op">)</span>
<span class="co">#&gt; [1] 2417</span>
<span class="va">core.glist</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">graphData</span>
<span class="co">#&gt; $edgemode</span>
<span class="co">#&gt; [1] "undirected"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $ComplexID</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $ComplexName</span>
<span class="co">#&gt; [1] "BCL6-HDAC4 complex"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $GO.ID</span>
<span class="co">#&gt; [1] "GO:0006265" "GO:0045892" "GO:0051276" "GO:0030183" "GO:0005634"</span>
<span class="co">#&gt; [6] "GO:0016575"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $PubMed.ID</span>
<span class="co">#&gt; [1] 11929873</span>
<span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/nodeData-methods.html" class="external-link">nodeData</a></span><span class="op">(</span><span class="va">core.glist</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt; $P41182</span>
<span class="co">#&gt; $P41182$ENTREZID</span>
<span class="co">#&gt; [1] "604"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $P41182$SYMBOL</span>
<span class="co">#&gt; [1] "BCL6"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $P56524</span>
<span class="co">#&gt; $P56524$ENTREZID</span>
<span class="co">#&gt; [1] "9759"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $P56524$SYMBOL</span>
<span class="co">#&gt; [1] "HDAC4"</span></code></pre></div>
<p>Note that we can easily convert a <a href="https://bioconductor.org/packages/graph" class="external-link">graph</a> object into an
<a href="https://cran.r-project.org/package=igraph" class="external-link">igraph</a> object
using <code><a href="https://rdrr.io/pkg/igraph/man/graph_from_graphnel.html" class="external-link">igraph::graph_from_graphnel</a></code>.</p>
</div>
</div>
<div class="section level3">
<h3 id="cnv-data">CNV data<a class="anchor" aria-label="anchor" href="#cnv-data"></a>
</h3>
<div class="section level4">
<h4 id="hek293t-cells">HEK293T cells<a class="anchor" aria-label="anchor" href="#hek293t-cells"></a>
</h4>
<p>Genomic data from whole-genome sequencing for six different lineages
of the human embryonic kidney HEK293 cell line can be obtained from <a href="http://hek293genome.org" class="external-link">hek293genome.org</a>.</p>
<p>This includes copy number variation (CNV) data for the 293T cell
line. Available CNV tracks include (i) CNV regions inferred from
sequencing read-depth analysis, and (ii) CNV regions inferred from
Illumina SNP arrays.</p>
<p>Here, we obtain CNV segments obtained from applying a hidden Markov
model (HMM) to sequencing-inferred copy numbers in 2kbp windows. More
details on how copy numbers were calculated can be obtained from the <a href="https://doi.org/10.1038/ncomms5767" class="external-link">primary publication</a>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cnv.hmm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getHEK293GenomeTrack.html">getHEK293GenomeTrack</a></span><span class="op">(</span>track <span class="op">=</span> <span class="st">"cnv.hmm"</span>, cell.line <span class="op">=</span> <span class="st">"293T"</span><span class="op">)</span>
<span class="co">#&gt; Using cached version from 2023-02-26 15:10:50</span>
<span class="va">cnv.hmm</span>
<span class="co">#&gt; GRanges object with 12382 ranges and 1 metadata column:</span>
<span class="co">#&gt;           seqnames              ranges strand |     score</span>
<span class="co">#&gt;              &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;</span>
<span class="co">#&gt;       [1]     chr1       823231-829231      * |      3.26</span>
<span class="co">#&gt;       [2]     chr1       835231-913231      * |      3.08</span>
<span class="co">#&gt;       [3]     chr1      923231-1063231      * |      3.20</span>
<span class="co">#&gt;       [4]     chr1     1079231-1213231      * |      3.21</span>
<span class="co">#&gt;       [5]     chr1     1223231-1399231      * |      3.27</span>
<span class="co">#&gt;       ...      ...                 ...    ... .       ...</span>
<span class="co">#&gt;   [12378]     chrX 154750237-154762237      * |      3.96</span>
<span class="co">#&gt;   [12379]     chrX 154778237-154780237      * |      4.02</span>
<span class="co">#&gt;   [12380]     chrX 154802237-154822237      * |      3.79</span>
<span class="co">#&gt;   [12381]     chrX 154842237-154846237      * |      3.85</span>
<span class="co">#&gt;   [12382]     chrM             0-12000      * |      8.82</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 24 sequences from hg18 genome; no seqlengths</span></code></pre></div>
<p>See also the <a href="https://bioconductor.org/packages/release/data/experiment/vignettes/BioPlex/inst/doc/BasicChecks.html" class="external-link">data
checks vignette, Section 5</a> for an exploration of the agreement
between inferred copy numbers from both assay types (SNP arrays
vs. sequencing).</p>
</div>
</div>
<div class="section level3">
<h3 id="transcriptome-data">Transcriptome data<a class="anchor" aria-label="anchor" href="#transcriptome-data"></a>
</h3>
<div class="section level4">
<h4 id="hek293t-cells-1">HEK293T cells<a class="anchor" aria-label="anchor" href="#hek293t-cells-1"></a>
</h4>
<div class="section level5">
<h5 id="gse122425">GSE122425<a class="anchor" aria-label="anchor" href="#gse122425"></a>
</h5>
<p>Obtain transcriptome data for 293T cells from GEO dataset: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122425" class="external-link">GSE122425</a>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getGSE122425.html">getGSE122425</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Using cached version from 2023-02-26 15:11:00</span>
<span class="va">se</span>
<span class="co">#&gt; class: SummarizedExperiment </span>
<span class="co">#&gt; dim: 57905 6 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(2): raw rpkm</span>
<span class="co">#&gt; rownames(57905): ENSG00000223972 ENSG00000227232 ... ENSG00000231514</span>
<span class="co">#&gt;   ENSG00000235857</span>
<span class="co">#&gt; rowData names(4): SYMBOL KO GO length</span>
<span class="co">#&gt; colnames(6): GSM3466389 GSM3466390 ... GSM3466393 GSM3466394</span>
<span class="co">#&gt; colData names(41): title geo_accession ... passages.ch1 strain.ch1</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"raw"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;                 GSM3466389 GSM3466390 GSM3466391 GSM3466392 GSM3466393</span>
<span class="co">#&gt; ENSG00000223972          1          2          2          0          0</span>
<span class="co">#&gt; ENSG00000227232        732        690        804        705        812</span>
<span class="co">#&gt; ENSG00000243485          0          0          2          0          0</span>
<span class="co">#&gt; ENSG00000237613          0          0          0          0          0</span>
<span class="co">#&gt; ENSG00000268020          0          0          0          0          0</span>
<span class="co">#&gt; ENSG00000240361          0          0          0          0          0</span>
<span class="co">#&gt;                 GSM3466394</span>
<span class="co">#&gt; ENSG00000223972          2</span>
<span class="co">#&gt; ENSG00000227232       1121</span>
<span class="co">#&gt; ENSG00000243485          0</span>
<span class="co">#&gt; ENSG00000237613          0</span>
<span class="co">#&gt; ENSG00000268020          0</span>
<span class="co">#&gt; ENSG00000240361          1</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"rpkm"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;                 GSM3466389 GSM3466390 GSM3466391 GSM3466392 GSM3466393</span>
<span class="co">#&gt; ENSG00000223972       0.01       0.01       0.01       0.00       0.00</span>
<span class="co">#&gt; ENSG00000227232       5.43       5.07       5.39       4.77       5.21</span>
<span class="co">#&gt; ENSG00000243485       0.00       0.00       0.04       0.00       0.00</span>
<span class="co">#&gt; ENSG00000237613       0.00       0.00       0.00       0.00       0.00</span>
<span class="co">#&gt; ENSG00000268020       0.00       0.00       0.00       0.00       0.00</span>
<span class="co">#&gt; ENSG00000240361       0.00       0.00       0.00       0.00       0.00</span>
<span class="co">#&gt;                 GSM3466394</span>
<span class="co">#&gt; ENSG00000223972       0.01</span>
<span class="co">#&gt; ENSG00000227232       6.80</span>
<span class="co">#&gt; ENSG00000243485       0.00</span>
<span class="co">#&gt; ENSG00000237613       0.00</span>
<span class="co">#&gt; ENSG00000268020       0.00</span>
<span class="co">#&gt; ENSG00000240361       0.02</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 6 rows and 41 columns</span>
<span class="co">#&gt;                    title geo_accession                status submission_date</span>
<span class="co">#&gt;              &lt;character&gt;   &lt;character&gt;           &lt;character&gt;     &lt;character&gt;</span>
<span class="co">#&gt; GSM3466389       WT rep1    GSM3466389 Public on Nov 16 2018     Nov 12 2018</span>
<span class="co">#&gt; GSM3466390       WT rep2    GSM3466390 Public on Nov 16 2018     Nov 12 2018</span>
<span class="co">#&gt; GSM3466391       WT rep3    GSM3466391 Public on Nov 16 2018     Nov 12 2018</span>
<span class="co">#&gt; GSM3466392 NSUN2-KO rep1    GSM3466392 Public on Nov 16 2018     Nov 12 2018</span>
<span class="co">#&gt; GSM3466393 NSUN2-KO rep2    GSM3466393 Public on Nov 16 2018     Nov 12 2018</span>
<span class="co">#&gt; GSM3466394 NSUN2-KO rep3    GSM3466394 Public on Nov 16 2018     Nov 12 2018</span>
<span class="co">#&gt;            last_update_date        type channel_count source_name_ch1</span>
<span class="co">#&gt;                 &lt;character&gt; &lt;character&gt;   &lt;character&gt;     &lt;character&gt;</span>
<span class="co">#&gt; GSM3466389      Nov 16 2018         SRA             1          kidney</span>
<span class="co">#&gt; GSM3466390      Nov 16 2018         SRA             1          kidney</span>
<span class="co">#&gt; GSM3466391      Nov 16 2018         SRA             1          kidney</span>
<span class="co">#&gt; GSM3466392      Nov 16 2018         SRA             1          kidney</span>
<span class="co">#&gt; GSM3466393      Nov 16 2018         SRA             1          kidney</span>
<span class="co">#&gt; GSM3466394      Nov 16 2018         SRA             1          kidney</span>
<span class="co">#&gt;            organism_ch1 characteristics_ch1 characteristics_ch1.1</span>
<span class="co">#&gt;             &lt;character&gt;         &lt;character&gt;           &lt;character&gt;</span>
<span class="co">#&gt; GSM3466389 Homo sapiens      strain: HEK293        passages: 8-12</span>
<span class="co">#&gt; GSM3466390 Homo sapiens      strain: HEK293        passages: 8-12</span>
<span class="co">#&gt; GSM3466391 Homo sapiens      strain: HEK293        passages: 8-12</span>
<span class="co">#&gt; GSM3466392 Homo sapiens      strain: HEK293        passages: 8-12</span>
<span class="co">#&gt; GSM3466393 Homo sapiens      strain: HEK293        passages: 8-12</span>
<span class="co">#&gt; GSM3466394 Homo sapiens      strain: HEK293        passages: 8-12</span>
<span class="co">#&gt;            treatment_protocol_ch1    growth_protocol_ch1 molecule_ch1</span>
<span class="co">#&gt;                       &lt;character&gt;            &lt;character&gt;  &lt;character&gt;</span>
<span class="co">#&gt; GSM3466389 The NSUN2-deficient .. Cells were grown in ..    polyA RNA</span>
<span class="co">#&gt; GSM3466390 The NSUN2-deficient .. Cells were grown in ..    polyA RNA</span>
<span class="co">#&gt; GSM3466391 The NSUN2-deficient .. Cells were grown in ..    polyA RNA</span>
<span class="co">#&gt; GSM3466392 The NSUN2-deficient .. Cells were grown in ..    polyA RNA</span>
<span class="co">#&gt; GSM3466393 The NSUN2-deficient .. Cells were grown in ..    polyA RNA</span>
<span class="co">#&gt; GSM3466394 The NSUN2-deficient .. Cells were grown in ..    polyA RNA</span>
<span class="co">#&gt;              extract_protocol_ch1 extract_protocol_ch1.1   taxid_ch1</span>
<span class="co">#&gt;                       &lt;character&gt;            &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt; GSM3466389 Total RNA of cells w.. Next generation sequ..        9606</span>
<span class="co">#&gt; GSM3466390 Total RNA of cells w.. Next generation sequ..        9606</span>
<span class="co">#&gt; GSM3466391 Total RNA of cells w.. Next generation sequ..        9606</span>
<span class="co">#&gt; GSM3466392 Total RNA of cells w.. Next generation sequ..        9606</span>
<span class="co">#&gt; GSM3466393 Total RNA of cells w.. Next generation sequ..        9606</span>
<span class="co">#&gt; GSM3466394 Total RNA of cells w.. Next generation sequ..        9606</span>
<span class="co">#&gt;                   data_processing      data_processing.1      data_processing.2</span>
<span class="co">#&gt;                       &lt;character&gt;            &lt;character&gt;            &lt;character&gt;</span>
<span class="co">#&gt; GSM3466389 Bcl2fastq (v2.17.1.1.. 3’ adaptor-trimming .. The high quality tri..</span>
<span class="co">#&gt; GSM3466390 Bcl2fastq (v2.17.1.1.. 3’ adaptor-trimming .. The high quality tri..</span>
<span class="co">#&gt; GSM3466391 Bcl2fastq (v2.17.1.1.. 3’ adaptor-trimming .. The high quality tri..</span>
<span class="co">#&gt; GSM3466392 Bcl2fastq (v2.17.1.1.. 3’ adaptor-trimming .. The high quality tri..</span>
<span class="co">#&gt; GSM3466393 Bcl2fastq (v2.17.1.1.. 3’ adaptor-trimming .. The high quality tri..</span>
<span class="co">#&gt; GSM3466394 Bcl2fastq (v2.17.1.1.. 3’ adaptor-trimming .. The high quality tri..</span>
<span class="co">#&gt;                 data_processing.3      data_processing.4  data_processing.5</span>
<span class="co">#&gt;                       &lt;character&gt;            &lt;character&gt;        &lt;character&gt;</span>
<span class="co">#&gt; GSM3466389 Reads Per Kilobase o.. Differential express.. Genome_build: HG19</span>
<span class="co">#&gt; GSM3466390 Reads Per Kilobase o.. Differential express.. Genome_build: HG19</span>
<span class="co">#&gt; GSM3466391 Reads Per Kilobase o.. Differential express.. Genome_build: HG19</span>
<span class="co">#&gt; GSM3466392 Reads Per Kilobase o.. Differential express.. Genome_build: HG19</span>
<span class="co">#&gt; GSM3466393 Reads Per Kilobase o.. Differential express.. Genome_build: HG19</span>
<span class="co">#&gt; GSM3466394 Reads Per Kilobase o.. Differential express.. Genome_build: HG19</span>
<span class="co">#&gt;                 data_processing.6 platform_id contact_name   contact_institute</span>
<span class="co">#&gt;                       &lt;character&gt; &lt;character&gt;  &lt;character&gt;         &lt;character&gt;</span>
<span class="co">#&gt; GSM3466389 Supplementary_files_..    GPL11154    Zhen,,Sun Yangzhou University</span>
<span class="co">#&gt; GSM3466390 Supplementary_files_..    GPL11154    Zhen,,Sun Yangzhou University</span>
<span class="co">#&gt; GSM3466391 Supplementary_files_..    GPL11154    Zhen,,Sun Yangzhou University</span>
<span class="co">#&gt; GSM3466392 Supplementary_files_..    GPL11154    Zhen,,Sun Yangzhou University</span>
<span class="co">#&gt; GSM3466393 Supplementary_files_..    GPL11154    Zhen,,Sun Yangzhou University</span>
<span class="co">#&gt; GSM3466394 Supplementary_files_..    GPL11154    Zhen,,Sun Yangzhou University</span>
<span class="co">#&gt;            contact_address contact_city contact_zip.postal_code contact_country</span>
<span class="co">#&gt;                &lt;character&gt;  &lt;character&gt;             &lt;character&gt;     &lt;character&gt;</span>
<span class="co">#&gt; GSM3466389          Wenhui     Yangzhou                  225009           China</span>
<span class="co">#&gt; GSM3466390          Wenhui     Yangzhou                  225009           China</span>
<span class="co">#&gt; GSM3466391          Wenhui     Yangzhou                  225009           China</span>
<span class="co">#&gt; GSM3466392          Wenhui     Yangzhou                  225009           China</span>
<span class="co">#&gt; GSM3466393          Wenhui     Yangzhou                  225009           China</span>
<span class="co">#&gt; GSM3466394          Wenhui     Yangzhou                  225009           China</span>
<span class="co">#&gt;            data_row_count    instrument_model library_selection library_source</span>
<span class="co">#&gt;               &lt;character&gt;         &lt;character&gt;       &lt;character&gt;    &lt;character&gt;</span>
<span class="co">#&gt; GSM3466389              0 Illumina HiSeq 2000              cDNA transcriptomic</span>
<span class="co">#&gt; GSM3466390              0 Illumina HiSeq 2000              cDNA transcriptomic</span>
<span class="co">#&gt; GSM3466391              0 Illumina HiSeq 2000              cDNA transcriptomic</span>
<span class="co">#&gt; GSM3466392              0 Illumina HiSeq 2000              cDNA transcriptomic</span>
<span class="co">#&gt; GSM3466393              0 Illumina HiSeq 2000              cDNA transcriptomic</span>
<span class="co">#&gt; GSM3466394              0 Illumina HiSeq 2000              cDNA transcriptomic</span>
<span class="co">#&gt;            library_strategy               relation             relation.1</span>
<span class="co">#&gt;                 &lt;character&gt;            &lt;character&gt;            &lt;character&gt;</span>
<span class="co">#&gt; GSM3466389          RNA-Seq BioSample: https://w.. SRA: https://www.ncb..</span>
<span class="co">#&gt; GSM3466390          RNA-Seq BioSample: https://w.. SRA: https://www.ncb..</span>
<span class="co">#&gt; GSM3466391          RNA-Seq BioSample: https://w.. SRA: https://www.ncb..</span>
<span class="co">#&gt; GSM3466392          RNA-Seq BioSample: https://w.. SRA: https://www.ncb..</span>
<span class="co">#&gt; GSM3466393          RNA-Seq BioSample: https://w.. SRA: https://www.ncb..</span>
<span class="co">#&gt; GSM3466394          RNA-Seq BioSample: https://w.. SRA: https://www.ncb..</span>
<span class="co">#&gt;            supplementary_file_1 passages.ch1  strain.ch1</span>
<span class="co">#&gt;                     &lt;character&gt;  &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt; GSM3466389                 NONE         8-12      HEK293</span>
<span class="co">#&gt; GSM3466390                 NONE         8-12      HEK293</span>
<span class="co">#&gt; GSM3466391                 NONE         8-12      HEK293</span>
<span class="co">#&gt; GSM3466392                 NONE         8-12      HEK293</span>
<span class="co">#&gt; GSM3466393                 NONE         8-12      HEK293</span>
<span class="co">#&gt; GSM3466394                 NONE         8-12      HEK293</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 57905 rows and 4 columns</span>
<span class="co">#&gt;                      SYMBOL          KO          GO    length</span>
<span class="co">#&gt;                 &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;integer&gt;</span>
<span class="co">#&gt; ENSG00000223972     DDX11L1      K11273           _      2544</span>
<span class="co">#&gt; ENSG00000227232      WASH7P      K18461           _     15444</span>
<span class="co">#&gt; ENSG00000243485  MIR1302-10           _           _      1556</span>
<span class="co">#&gt; ENSG00000237613     FAM138A           _           _      1528</span>
<span class="co">#&gt; ENSG00000268020      OR4G4P      K04257           _      2464</span>
<span class="co">#&gt; ...                     ...         ...         ...       ...</span>
<span class="co">#&gt; ENSG00000224240     CYCSP49      K08738           _       319</span>
<span class="co">#&gt; ENSG00000227629  SLC25A15P1      K15101           _      4960</span>
<span class="co">#&gt; ENSG00000237917     PARP4P1      K10798           _     39802</span>
<span class="co">#&gt; ENSG00000231514     FAM58CP           _           _       640</span>
<span class="co">#&gt; ENSG00000235857     CTBP2P1      K04496           _       245</span></code></pre></div>
<p>The dataset includes three wild type samples and three NSUN2 knockout
samples.</p>
<p>See also the <a href="https://bioconductor.org/packages/release/data/experiment/vignettes/BioPlex/inst/doc/BasicChecks.html" class="external-link">data
checks vignette, Section 7</a> for an exploration of the relationship
between expression level and the frequency of a protein being detected
as prey.</p>
</div>
</div>
<div class="section level4">
<h4 id="hct116-cells">HCT116 cells<a class="anchor" aria-label="anchor" href="#hct116-cells"></a>
</h4>
<div class="section level5">
<h5 id="cancer-cell-line-encyclopedia-ccle">Cancer Cell Line Encyclopedia (CCLE)<a class="anchor" aria-label="anchor" href="#cancer-cell-line-encyclopedia-ccle"></a>
</h5>
<p>RNA-seq data for 934 cancer cell lines (incl. HCT116) from the <a href="https://portals.broadinstitute.org/ccle" class="external-link">Cancer Cell Line
Encyclopedia</a> is available from the <a href="https://www.ebi.ac.uk/gxa" class="external-link">ArrayExpress-ExpressionAtlas</a>
(Accession: <a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-2770" class="external-link">E-MTAB-2770</a>).</p>
<p>The data can be obtained as a <code>SummarizedExperiment</code> using
the <a href="https://bioconductor.org/packages/ExpressionAtlas" class="external-link">ExpressionAtlas</a>
package.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ccle.trans</span> <span class="op">&lt;-</span> <span class="fu">ExpressionAtlas</span><span class="fu">::</span><span class="fu">getAtlasExperiment</span><span class="op">(</span><span class="st">"E-MTAB-2770"</span><span class="op">)</span></code></pre></div>
<p>See also the <a href="https://ccb-hms.github.io/BioPlexAnalysis/articles/TranscriptomeProteome.html">Transcriptome-Proteome
analysis vignette</a> for further exploration of the correlation between
CCLE HCT116 transcript and protein expression.</p>
</div>
<div class="section level5">
<h5 id="klijn-et-al--2015">Klijn et al., 2015<a class="anchor" aria-label="anchor" href="#klijn-et-al--2015"></a>
</h5>
<p>RNA-seq data of 675 commonly used human cancer cell lines
(incl. HCT116) from <a href="https://pubmed.ncbi.nlm.nih.gov/25485619" class="external-link">Klijn et al., 2015</a>
is available from the <a href="https://www.ebi.ac.uk/gxa" class="external-link">ArrayExpress-ExpressionAtlas</a>
(Accession: <a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-2706" class="external-link">E-MTAB-2706</a>)</p>
<p>The data can be obtained as a <code>SummarizedExperiment</code> using
the <a href="https://bioconductor.org/packages/ExpressionAtlas" class="external-link">ExpressionAtlas</a>
package.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">klijn</span> <span class="op">&lt;-</span> <span class="fu">ExpressionAtlas</span><span class="fu">::</span><span class="fu">getAtlasExperiment</span><span class="op">(</span><span class="st">"E-MTAB-2706"</span><span class="op">)</span></code></pre></div>
<p>See also the <a href="https://ccb-hms.github.io/BioPlexAnalysis/articles/TranscriptomeProteome.html">Transcriptome-Proteome
analysis vignette</a> for further exploration of differential transcript
and protein expression between 293T and HCT116 cells.</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="splicing-data">Splicing data<a class="anchor" aria-label="anchor" href="#splicing-data"></a>
</h3>
<p>For the inference of differential exon usage between cell lines, raw
RNA-seq read counts on exon level can be obtained from <a href="https://bioconductor.org/packages/ExperimentHub" class="external-link">ExperimentHub</a>.</p>
<p>RNA-seq data for 293T cells was obtained from GEO accession <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122633" class="external-link">GSE122633</a>
and RNA-seq data for HCT116 cells was obtained from GEO accession <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE52429" class="external-link">GSE52429</a>.</p>
<p>The data can be obtained as a <code>DEXSeqDataSet</code> which is a
<code>SummarizedExperiment</code>-derivative and can be accessed and
manipulated very much like a <code>DESeqDataSet</code>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">eh</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"BioPlex"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; ExperimentHub with 1 record</span>
<span class="co">#&gt; # snapshotDate(): 2022-10-31</span>
<span class="co">#&gt; # names(): EH7563</span>
<span class="co">#&gt; # package(): BioPlex</span>
<span class="co">#&gt; # $dataprovider: University of Southern California</span>
<span class="co">#&gt; # $species: Homo sapiens</span>
<span class="co">#&gt; # $rdataclass: DEXSeqDataSet</span>
<span class="co">#&gt; # $rdatadateadded: 2022-07-15</span>
<span class="co">#&gt; # $title: HEK293T_HCT116_exon_counts</span>
<span class="co">#&gt; # $description: Exon count matrix</span>
<span class="co">#&gt; # $taxonomyid: 9606</span>
<span class="co">#&gt; # $genome: hg38</span>
<span class="co">#&gt; # $sourcetype: Zip</span>
<span class="co">#&gt; # $sourceurl: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE52429</span>
<span class="co">#&gt; # $sourcesize: NA</span>
<span class="co">#&gt; # $tags: c("CellCulture", "ColonCancerData", "ExpressionData",</span>
<span class="co">#&gt; #   "Genome", "GEO", "Homo_sapiens_Data", "MassSpectrometryData",</span>
<span class="co">#&gt; #   "Proteome", "ReproducibleResearch", "RNASeqData") </span>
<span class="co">#&gt; # retrieve record with 'object[["EH7563"]]'</span>
<span class="va">dex</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">"EH7563"</span><span class="op">]</span><span class="op">]</span>
<span class="va">dex</span>
<span class="co">#&gt; class: DEXSeqDataSet </span>
<span class="co">#&gt; dim: 607086 10 </span>
<span class="co">#&gt; metadata(1): version</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(607086): ENSG00000000003:E001 ENSG00000000003:E002 ...</span>
<span class="co">#&gt;   ENSG00000288723:E005 ENSG00000288723:E006</span>
<span class="co">#&gt; rowData names(8): featureID groupID ... chromosome biotype</span>
<span class="co">#&gt; colnames: NULL</span>
<span class="co">#&gt; colData names(9): sample sampleName ... exon sizeFactor</span></code></pre></div>
<p>We take a closer look at the sample annotation, the counts for each
exon for both cell lines, and the genomic coordinates and additional
annotation for each exon.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">DEXSeq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DEXSeq/man/accessors.html" class="external-link">sampleAnnotation</a></span><span class="op">(</span><span class="va">dex</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 5 rows and 8 columns</span>
<span class="co">#&gt;       sample           sampleName      GEO_ID Platform_ID         End</span>
<span class="co">#&gt;     &lt;factor&gt;          &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt; 1 GSM1266733 HCT116 RNA-seq repPF    GSE52429    GPL11154      paired</span>
<span class="co">#&gt; 2 GSM1266734 HCT116 RNA-seq repPJ    GSE52429    GPL11154      paired</span>
<span class="co">#&gt; 3 GSM3476820     UnTfx1_S7_R1_001   GSE122633    GPL18573      single</span>
<span class="co">#&gt; 4 GSM3476821     UnTfx2_S8_R1_001   GSE122633    GPL18573      single</span>
<span class="co">#&gt; 5 GSM3476822     UnTfx3_S9_R1_001   GSE122633    GPL18573      single</span>
<span class="co">#&gt;      Stranded    cellLine sizeFactor</span>
<span class="co">#&gt;   &lt;character&gt; &lt;character&gt;  &lt;numeric&gt;</span>
<span class="co">#&gt; 1  unstranded      HCT116   1.336168</span>
<span class="co">#&gt; 2  unstranded      HCT116   1.383283</span>
<span class="co">#&gt; 3    stranded     HEK293T   0.860023</span>
<span class="co">#&gt; 4    stranded     HEK293T   0.803370</span>
<span class="co">#&gt; 5    stranded     HEK293T   0.848761</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">DEXSeq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DEXSeq/man/accessors.html" class="external-link">featureCounts</a></span><span class="op">(</span><span class="va">dex</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;                      GSM1266733 GSM1266734 GSM3476820 GSM3476821 GSM3476822</span>
<span class="co">#&gt; ENSG00000000003:E001          2          0          2          0          2</span>
<span class="co">#&gt; ENSG00000000003:E002       1249        939       2713       3321       3323</span>
<span class="co">#&gt; ENSG00000000003:E003        399        287        907       1050       1039</span>
<span class="co">#&gt; ENSG00000000003:E004          5          2          6          3          0</span>
<span class="co">#&gt; ENSG00000000003:E005        313        239        386        486        476</span>
<span class="co">#&gt; ENSG00000000003:E006        243        200        190        239        235</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">rowRanges</a></span><span class="op">(</span><span class="va">dex</span><span class="op">)</span>
<span class="co">#&gt; GRanges object with 607086 ranges and 8 metadata columns:</span>
<span class="co">#&gt;                        seqnames              ranges strand |   featureID</span>
<span class="co">#&gt;                           &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">#&gt;   ENSG00000000003:E001        X           100627108      - |        E001</span>
<span class="co">#&gt;   ENSG00000000003:E002        X 100627109-100629986      - |        E002</span>
<span class="co">#&gt;   ENSG00000000003:E003        X 100630759-100630866      - |        E003</span>
<span class="co">#&gt;   ENSG00000000003:E004        X 100632063-100632068      - |        E004</span>
<span class="co">#&gt;   ENSG00000000003:E005        X 100632485-100632540      - |        E005</span>
<span class="co">#&gt;                    ...      ...                 ...    ... .         ...</span>
<span class="co">#&gt;   ENSG00000288723:E002        1 241723858-241723938      - |        E002</span>
<span class="co">#&gt;   ENSG00000288723:E003        1 241728735-241728834      - |        E003</span>
<span class="co">#&gt;   ENSG00000288723:E004        1 241742059-241742106      - |        E004</span>
<span class="co">#&gt;   ENSG00000288723:E005        1 241765062-241765184      - |        E005</span>
<span class="co">#&gt;   ENSG00000288723:E006        1 241847969-241848128      - |        E006</span>
<span class="co">#&gt;                                groupID exonBaseMean exonBaseVar</span>
<span class="co">#&gt;                            &lt;character&gt;    &lt;numeric&gt;   &lt;numeric&gt;</span>
<span class="co">#&gt;   ENSG00000000003:E001 ENSG00000000003          1.2         1.2</span>
<span class="co">#&gt;   ENSG00000000003:E002 ENSG00000000003       2309.0   1304014.0</span>
<span class="co">#&gt;   ENSG00000000003:E003 ENSG00000000003        736.4    133703.8</span>
<span class="co">#&gt;   ENSG00000000003:E004 ENSG00000000003          3.2         5.7</span>
<span class="co">#&gt;   ENSG00000000003:E005 ENSG00000000003        380.0     11214.5</span>
<span class="co">#&gt;                    ...             ...          ...         ...</span>
<span class="co">#&gt;   ENSG00000288723:E002 ENSG00000288723          1.0         1.5</span>
<span class="co">#&gt;   ENSG00000288723:E003 ENSG00000288723          1.8         3.7</span>
<span class="co">#&gt;   ENSG00000288723:E004 ENSG00000288723          2.0         6.0</span>
<span class="co">#&gt;   ENSG00000288723:E005 ENSG00000288723          1.8         6.2</span>
<span class="co">#&gt;   ENSG00000288723:E006 ENSG00000288723          0.2         0.2</span>
<span class="co">#&gt;                                            transcripts      symbol  chromosome</span>
<span class="co">#&gt;                                                 &lt;list&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt;   ENSG00000000003:E001                 ENST00000373020      TSPAN6           X</span>
<span class="co">#&gt;   ENSG00000000003:E002 ENST00000373020,ENST00000612152      TSPAN6           X</span>
<span class="co">#&gt;   ENSG00000000003:E003 ENST00000373020,ENST00000612152      TSPAN6           X</span>
<span class="co">#&gt;   ENSG00000000003:E004                 ENST00000614008      TSPAN6           X</span>
<span class="co">#&gt;   ENSG00000000003:E005 ENST00000373020,ENST00000614008      TSPAN6           X</span>
<span class="co">#&gt;                    ...                             ...         ...         ...</span>
<span class="co">#&gt;   ENSG00000288723:E002                 ENST00000684005                       1</span>
<span class="co">#&gt;   ENSG00000288723:E003                 ENST00000684005                       1</span>
<span class="co">#&gt;   ENSG00000288723:E004                 ENST00000684005                       1</span>
<span class="co">#&gt;   ENSG00000288723:E005                 ENST00000684005                       1</span>
<span class="co">#&gt;   ENSG00000288723:E006                 ENST00000684005                       1</span>
<span class="co">#&gt;                               biotype</span>
<span class="co">#&gt;                           &lt;character&gt;</span>
<span class="co">#&gt;   ENSG00000000003:E001 protein_coding</span>
<span class="co">#&gt;   ENSG00000000003:E002 protein_coding</span>
<span class="co">#&gt;   ENSG00000000003:E003 protein_coding</span>
<span class="co">#&gt;   ENSG00000000003:E004 protein_coding</span>
<span class="co">#&gt;   ENSG00000000003:E005 protein_coding</span>
<span class="co">#&gt;                    ...            ...</span>
<span class="co">#&gt;   ENSG00000288723:E002         lncRNA</span>
<span class="co">#&gt;   ENSG00000288723:E003         lncRNA</span>
<span class="co">#&gt;   ENSG00000288723:E004         lncRNA</span>
<span class="co">#&gt;   ENSG00000288723:E005         lncRNA</span>
<span class="co">#&gt;   ENSG00000288723:E006         lncRNA</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 47 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="proteome-data">Proteome data<a class="anchor" aria-label="anchor" href="#proteome-data"></a>
</h3>
<div class="section level4">
<h4 id="ccle">CCLE<a class="anchor" aria-label="anchor" href="#ccle"></a>
</h4>
<p>Pull the <a href="https://doi.org/10.1016/j.cell.2019.12.023" class="external-link">CCLE
proteome data</a> from ExperimentHub. The dataset profiles 12,755
proteins by mass spectrometry across 375 cancer cell lines.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">eh</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gygi"</span>, <span class="st">"depmap"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; ExperimentHub with 1 record</span>
<span class="co">#&gt; # snapshotDate(): 2022-10-31</span>
<span class="co">#&gt; # names(): EH3459</span>
<span class="co">#&gt; # package(): depmap</span>
<span class="co">#&gt; # $dataprovider: Broad Institute</span>
<span class="co">#&gt; # $species: Homo sapiens</span>
<span class="co">#&gt; # $rdataclass: tibble</span>
<span class="co">#&gt; # $rdatadateadded: 2020-05-19</span>
<span class="co">#&gt; # $title: proteomic_20Q2</span>
<span class="co">#&gt; # $description: Quantitative profiling of 12399 proteins in 375 cell lines, ...</span>
<span class="co">#&gt; # $taxonomyid: 9606</span>
<span class="co">#&gt; # $genome: </span>
<span class="co">#&gt; # $sourcetype: CSV</span>
<span class="co">#&gt; # $sourceurl: https://gygi.med.harvard.edu/sites/gygi.med.harvard.edu/files/...</span>
<span class="co">#&gt; # $sourcesize: NA</span>
<span class="co">#&gt; # $tags: c("ExperimentHub", "ExperimentData", "ReproducibleResearch",</span>
<span class="co">#&gt; #   "RepositoryData", "AssayDomainData", "CopyNumberVariationData",</span>
<span class="co">#&gt; #   "DiseaseModel", "CancerData", "BreastCancerData", "ColonCancerData",</span>
<span class="co">#&gt; #   "KidneyCancerData", "LeukemiaCancerData", "LungCancerData",</span>
<span class="co">#&gt; #   "OvarianCancerData", "ProstateCancerData", "OrganismData",</span>
<span class="co">#&gt; #   "Homo_sapiens_Data", "PackageTypeData", "SpecimenSource",</span>
<span class="co">#&gt; #   "CellCulture", "Genome", "Proteome", "StemCell", "Tissue") </span>
<span class="co">#&gt; # retrieve record with 'object[["EH3459"]]'</span>
<span class="va">ccle.prot</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">"EH3459"</span><span class="op">]</span><span class="op">]</span>
<span class="va">ccle.prot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">ccle.prot</span><span class="op">)</span></code></pre></div>
<p>Explore the data:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">ccle.prot</span><span class="op">)</span>
<span class="co">#&gt; [1] 4821390      12</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">ccle.prot</span><span class="op">)</span>
<span class="co">#&gt;  [1] "depmap_id"          "gene_name"          "entrez_id"         </span>
<span class="co">#&gt;  [4] "protein"            "protein_expression" "protein_id"        </span>
<span class="co">#&gt;  [7] "desc"               "group_id"           "uniprot"           </span>
<span class="co">#&gt; [10] "uniprot_acc"        "TenPx"              "cell_line"</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ccle.prot</span><span class="op">)</span>
<span class="co">#&gt;    depmap_id gene_name entrez_id                 protein protein_expression</span>
<span class="co">#&gt; 1 ACH-000849   SLC12A2      6558 MDAMB468_BREAST_TenPx01         2.11134846</span>
<span class="co">#&gt; 2 ACH-000441   SLC12A2      6558        SH4_SKIN_TenPx01         0.07046807</span>
<span class="co">#&gt; 3 ACH-000248   SLC12A2      6558    AU565_BREAST_TenPx01        -0.46392793</span>
<span class="co">#&gt; 4 ACH-000684   SLC12A2      6558    KMRC1_KIDNEY_TenPx01        -0.88364548</span>
<span class="co">#&gt; 5 ACH-000856   SLC12A2      6558    CAL51_BREAST_TenPx01         0.78856534</span>
<span class="co">#&gt; 6 ACH-000348   SLC12A2      6558   RPMI7951_SKIN_TenPx01        -0.91235198</span>
<span class="co">#&gt;              protein_id                                          desc group_id</span>
<span class="co">#&gt; 1 sp|P55011|S12A2_HUMAN S12A2_HUMAN Solute carrier family 12 member 2        0</span>
<span class="co">#&gt; 2 sp|P55011|S12A2_HUMAN S12A2_HUMAN Solute carrier family 12 member 2        0</span>
<span class="co">#&gt; 3 sp|P55011|S12A2_HUMAN S12A2_HUMAN Solute carrier family 12 member 2        0</span>
<span class="co">#&gt; 4 sp|P55011|S12A2_HUMAN S12A2_HUMAN Solute carrier family 12 member 2        0</span>
<span class="co">#&gt; 5 sp|P55011|S12A2_HUMAN S12A2_HUMAN Solute carrier family 12 member 2        0</span>
<span class="co">#&gt; 6 sp|P55011|S12A2_HUMAN S12A2_HUMAN Solute carrier family 12 member 2        0</span>
<span class="co">#&gt;       uniprot uniprot_acc   TenPx       cell_line</span>
<span class="co">#&gt; 1 S12A2_HUMAN      P55011 TenPx01 MDAMB468_BREAST</span>
<span class="co">#&gt; 2 S12A2_HUMAN      P55011 TenPx01        SH4_SKIN</span>
<span class="co">#&gt; 3 S12A2_HUMAN      P55011 TenPx01    AU565_BREAST</span>
<span class="co">#&gt; 4 S12A2_HUMAN      P55011 TenPx01    KMRC1_KIDNEY</span>
<span class="co">#&gt; 5 S12A2_HUMAN      P55011 TenPx01    CAL51_BREAST</span>
<span class="co">#&gt; 6 S12A2_HUMAN      P55011 TenPx01   RPMI7951_SKIN</span></code></pre></div>
<p>Restrict to HCT116:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ccle.prot.hct116</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">ccle.prot</span>, <span class="va">cell_line</span> <span class="op">==</span> <span class="st">"HCT116_LARGE_INTESTINE"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">ccle.prot.hct116</span><span class="op">)</span>
<span class="co">#&gt; [1] 12755    12</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ccle.prot.hct116</span><span class="op">)</span>
<span class="co">#&gt;       depmap_id gene_name entrez_id                        protein</span>
<span class="co">#&gt; 28   ACH-000971   SLC12A2      6558 HCT116_LARGE_INTESTINE_TenPx04</span>
<span class="co">#&gt; 406  ACH-000971    HOXD13      3239 HCT116_LARGE_INTESTINE_TenPx04</span>
<span class="co">#&gt; 784  ACH-000971     KDM1A     23028 HCT116_LARGE_INTESTINE_TenPx04</span>
<span class="co">#&gt; 1162 ACH-000971      SOX1      6656 HCT116_LARGE_INTESTINE_TenPx04</span>
<span class="co">#&gt; 1540 ACH-000971      SOX2      6657 HCT116_LARGE_INTESTINE_TenPx04</span>
<span class="co">#&gt; 1918 ACH-000971      SOX3      6658 HCT116_LARGE_INTESTINE_TenPx04</span>
<span class="co">#&gt;      protein_expression            protein_id</span>
<span class="co">#&gt; 28           -0.2422502 sp|P55011|S12A2_HUMAN</span>
<span class="co">#&gt; 406                  NA sp|P35453|HXD13_HUMAN</span>
<span class="co">#&gt; 784          -0.1941110 sp|O60341|KDM1A_HUMAN</span>
<span class="co">#&gt; 1162                 NA  sp|O00570|SOX1_HUMAN</span>
<span class="co">#&gt; 1540         -1.5306584  sp|P48431|SOX2_HUMAN</span>
<span class="co">#&gt; 1918                 NA  sp|P41225|SOX3_HUMAN</span>
<span class="co">#&gt;                                                    desc group_id     uniprot</span>
<span class="co">#&gt; 28        S12A2_HUMAN Solute carrier family 12 member 2        0 S12A2_HUMAN</span>
<span class="co">#&gt; 406                HXD13_HUMAN Homeobox protein Hox-D13        1 HXD13_HUMAN</span>
<span class="co">#&gt; 784  KDM1A_HUMAN Lysine-specific histone demethylase 1A        2 KDM1A_HUMAN</span>
<span class="co">#&gt; 1162              SOX1_HUMAN Transcription factor SOX-1        4  SOX1_HUMAN</span>
<span class="co">#&gt; 1540              SOX2_HUMAN Transcription factor SOX-2        4  SOX2_HUMAN</span>
<span class="co">#&gt; 1918              SOX3_HUMAN Transcription factor SOX-3        4  SOX3_HUMAN</span>
<span class="co">#&gt;      uniprot_acc   TenPx              cell_line</span>
<span class="co">#&gt; 28        P55011 TenPx04 HCT116_LARGE_INTESTINE</span>
<span class="co">#&gt; 406       P35453 TenPx04 HCT116_LARGE_INTESTINE</span>
<span class="co">#&gt; 784       O60341 TenPx04 HCT116_LARGE_INTESTINE</span>
<span class="co">#&gt; 1162      O00570 TenPx04 HCT116_LARGE_INTESTINE</span>
<span class="co">#&gt; 1540      P48431 TenPx04 HCT116_LARGE_INTESTINE</span>
<span class="co">#&gt; 1918      P41225 TenPx04 HCT116_LARGE_INTESTINE</span></code></pre></div>
<p>Or turn into a <code>SummarizedExperiment</code> for convenience (we
can restrict this to selected cell lines, but here we keep all cell
lines):</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ccleProteome2SummarizedExperiment.html">ccleProteome2SummarizedExperiment</a></span><span class="op">(</span><span class="va">ccle.prot</span>, cell.line <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>
<span class="co">#&gt;             22RV1         697       769P       786O      8305C</span>
<span class="co">#&gt; P55011  1.8112046 -0.01482998 -0.5658598 -1.2205591 -0.1713740</span>
<span class="co">#&gt; P35453         NA          NA -2.5433313         NA         NA</span>
<span class="co">#&gt; O60341 -0.3379936  0.37121437 -0.8170886 -0.9183874  0.1141192</span>
<span class="co">#&gt; O00570         NA          NA         NA -0.5043593         NA</span>
<span class="co">#&gt; P48431 -1.2617033          NA         NA -3.4006509         NA</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="st">"HCT116"</span><span class="op">]</span>
<span class="co">#&gt;     P55011     P35453     O60341     O00570     P48431 </span>
<span class="co">#&gt; -0.2422502         NA -0.1941110         NA -1.5306584</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 12755 rows and 2 columns</span>
<span class="co">#&gt;             SYMBOL  ENTREZID</span>
<span class="co">#&gt;        &lt;character&gt; &lt;numeric&gt;</span>
<span class="co">#&gt; P55011     SLC12A2      6558</span>
<span class="co">#&gt; P35453      HOXD13      3239</span>
<span class="co">#&gt; O60341       KDM1A     23028</span>
<span class="co">#&gt; O00570        SOX1      6656</span>
<span class="co">#&gt; P48431        SOX2      6657</span>
<span class="co">#&gt; ...            ...       ...</span>
<span class="co">#&gt; Q9Y258       CCL26     10344</span>
<span class="co">#&gt; P20292     ALOX5AP       241</span>
<span class="co">#&gt; Q9H1C7      CYSTM1     84418</span>
<span class="co">#&gt; Q99735       MGST2      4258</span>
<span class="co">#&gt; Q9P003       CNIH4     29097</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="relative-protein-expression-data-from-bioplex3-0">Relative protein expression data from BioPlex3.0<a class="anchor" aria-label="anchor" href="#relative-protein-expression-data-from-bioplex3-0"></a>
</h4>
<p>The <a href="https://doi.org/10.1016/j.cell.2021.04.011" class="external-link">BioPlex 3.0
publication</a>, Supplementary Table S4A, provides relative protein
expression data comparing 293T and HCT116 cells based on tandem mass tag
analysis.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.prot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getBioplexProteome.html">getBioplexProteome</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Using cached version from 2023-02-26 15:11:08</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>
<span class="co">#&gt;               HCT1      HCT2      HCT3      HCT4      HCT5</span>
<span class="co">#&gt; P0CG40    1.526690  3.479370  2.223500  2.258470  2.923410</span>
<span class="co">#&gt; Q8IXZ3-4  1.758790  1.610220  2.004360  1.800270  1.371470</span>
<span class="co">#&gt; P55011   12.570100 11.637500 12.495700 11.374500 12.377400</span>
<span class="co">#&gt; O60341    8.914910  9.677760  8.397850  8.745780  8.746410</span>
<span class="co">#&gt; O14654    0.196305  0.277787  0.425389  0.199624  0.491558</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 10 rows and 1 column</span>
<span class="co">#&gt;        cell.line</span>
<span class="co">#&gt;      &lt;character&gt;</span>
<span class="co">#&gt; HCT1      HCT116</span>
<span class="co">#&gt; HCT2      HCT116</span>
<span class="co">#&gt; HCT3      HCT116</span>
<span class="co">#&gt; HCT4      HCT116</span>
<span class="co">#&gt; HCT5      HCT116</span>
<span class="co">#&gt; HEK1        293T</span>
<span class="co">#&gt; HEK2        293T</span>
<span class="co">#&gt; HEK3        293T</span>
<span class="co">#&gt; HEK4        293T</span>
<span class="co">#&gt; HEK5        293T</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 9604 rows and 5 columns</span>
<span class="co">#&gt;             ENTREZID      SYMBOL nr.peptides log2ratio  adj.pvalue</span>
<span class="co">#&gt;          &lt;character&gt; &lt;character&gt;   &lt;integer&gt; &lt;numeric&gt;   &lt;numeric&gt;</span>
<span class="co">#&gt; P0CG40     100131390         SP9           1 -2.819071 6.66209e-08</span>
<span class="co">#&gt; Q8IXZ3-4      221833         SP8           3 -3.419888 6.94973e-07</span>
<span class="co">#&gt; P55011          6558     SLC12A2           4  0.612380 4.85602e-06</span>
<span class="co">#&gt; O60341         23028       KDM1A           7 -0.319695 5.08667e-04</span>
<span class="co">#&gt; O14654          8471        IRS4           4 -5.951096 1.45902e-06</span>
<span class="co">#&gt; ...              ...         ...         ...       ...         ...</span>
<span class="co">#&gt; Q9H6X4         80194     TMEM134           2 -0.379342 7.67195e-05</span>
<span class="co">#&gt; Q9BS91         55032     SLC35A5           1 -2.237634 8.75523e-05</span>
<span class="co">#&gt; Q9UKJ5         26511       CHIC2           1 -0.614932 1.78756e-03</span>
<span class="co">#&gt; Q9H3S5         93183        PIGM           1 -1.011397 8.91589e-06</span>
<span class="co">#&gt; Q8WYQ3        400916     CHCHD10           1  0.743852 1.17163e-03</span></code></pre></div>
<p>The data contains 5 replicates each for 293T and for HCT116 cells. As
a result of the data collection process, the data represent relative
protein abundance scaled to add up to 100% in each row.</p>
<p>See also the <a href="https://bioconductor.org/packages/release/data/experiment/vignettes/BioPlex/inst/doc/BasicChecks.html" class="external-link">data
checks vignette, Section 8</a> for a basic exploration of the annotated
differential expression measures.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="caching">Caching<a class="anchor" aria-label="anchor" href="#caching"></a>
</h2>
<p>Note that calling functions like <code>getCorum</code> or
<code>getBioPlex</code> with argument <code>cache = FALSE</code> will
automatically overwrite the corresponding object in your cache. It is
thus typically not required for a user to interact with the cache.</p>
<p>For more extended control of the cache, use from within R:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cache.dir</span> <span class="op">&lt;-</span> <span class="fu">tools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/tools/userdir.html" class="external-link">R_user_dir</a></span><span class="op">(</span><span class="st">"BioPlex"</span>, which <span class="op">=</span> <span class="st">"cache"</span><span class="op">)</span> 
<span class="va">bfc</span> <span class="op">&lt;-</span> <span class="fu">BiocFileCache</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">BiocFileCache</a></span><span class="op">(</span><span class="va">cache.dir</span><span class="op">)</span></code></pre></div>
<p>and then proceed as described in the <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/BiocFileCache/inst/doc/BiocFileCache.html#cleaning-or-removing-cache" class="external-link">BiocFileCache
vignette, Section 1.10</a> either via <code><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">cleanbfc()</a></code> to clean
or <code><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">removebfc()</a></code> to remove your cache.</p>
<p>To do a hard reset (use with caution!):</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocFileCache</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">removebfc</a></span><span class="op">(</span><span class="va">bfc</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="sessioninfo">SessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.2.0 (2022-04-22)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04.5 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span>
<span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">#&gt; [8] base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] depmap_1.12.0               dplyr_1.0.9                </span>
<span class="co">#&gt;  [3] DEXSeq_1.44.0               RColorBrewer_1.1-3         </span>
<span class="co">#&gt;  [5] DESeq2_1.38.3               BiocParallel_1.32.5        </span>
<span class="co">#&gt;  [7] AnnotationDbi_1.60.0        graph_1.76.0               </span>
<span class="co">#&gt;  [9] ExperimentHub_2.6.0         AnnotationHub_3.6.0        </span>
<span class="co">#&gt; [11] BiocFileCache_2.6.1         dbplyr_2.2.0               </span>
<span class="co">#&gt; [13] BioPlex_1.5.4               SummarizedExperiment_1.28.0</span>
<span class="co">#&gt; [15] Biobase_2.58.0              GenomicRanges_1.50.2       </span>
<span class="co">#&gt; [17] GenomeInfoDb_1.34.9         IRanges_2.32.0             </span>
<span class="co">#&gt; [19] S4Vectors_0.36.1            BiocGenerics_0.44.0        </span>
<span class="co">#&gt; [21] MatrixGenerics_1.10.0       matrixStats_0.62.0         </span>
<span class="co">#&gt; [23] BiocStyle_2.26.0           </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] colorspace_2.0-3              hwriter_1.3.2.1              </span>
<span class="co">#&gt;  [3] ellipsis_0.3.2                rprojroot_2.0.3              </span>
<span class="co">#&gt;  [5] XVector_0.38.0                fs_1.5.2                     </span>
<span class="co">#&gt;  [7] bit64_4.0.5                   interactiveDisplayBase_1.36.0</span>
<span class="co">#&gt;  [9] fansi_1.0.3                   xml2_1.3.3                   </span>
<span class="co">#&gt; [11] splines_4.2.0                 codetools_0.2-18             </span>
<span class="co">#&gt; [13] cachem_1.0.6                  geneplotter_1.76.0           </span>
<span class="co">#&gt; [15] knitr_1.39                    jsonlite_1.8.0               </span>
<span class="co">#&gt; [17] Rsamtools_2.14.0              annotate_1.76.0              </span>
<span class="co">#&gt; [19] png_0.1-7                     shiny_1.7.1                  </span>
<span class="co">#&gt; [21] BiocManager_1.30.18           compiler_4.2.0               </span>
<span class="co">#&gt; [23] httr_1.4.3                    assertthat_0.2.1             </span>
<span class="co">#&gt; [25] Matrix_1.4-1                  fastmap_1.1.0                </span>
<span class="co">#&gt; [27] cli_3.3.0                     later_1.3.0                  </span>
<span class="co">#&gt; [29] htmltools_0.5.2               prettyunits_1.1.1            </span>
<span class="co">#&gt; [31] tools_4.2.0                   gtable_0.3.0                 </span>
<span class="co">#&gt; [33] glue_1.6.2                    GenomeInfoDbData_1.2.9       </span>
<span class="co">#&gt; [35] rappdirs_0.3.3                Rcpp_1.0.8.3                 </span>
<span class="co">#&gt; [37] jquerylib_0.1.4               pkgdown_2.0.4                </span>
<span class="co">#&gt; [39] vctrs_0.4.1                   Biostrings_2.66.0            </span>
<span class="co">#&gt; [41] xfun_0.31                     stringr_1.4.0                </span>
<span class="co">#&gt; [43] mime_0.12                     lifecycle_1.0.1              </span>
<span class="co">#&gt; [45] statmod_1.4.36                XML_3.99-0.10                </span>
<span class="co">#&gt; [47] zlibbioc_1.44.0               scales_1.2.0                 </span>
<span class="co">#&gt; [49] ragg_1.2.2                    hms_1.1.1                    </span>
<span class="co">#&gt; [51] promises_1.2.0.1              parallel_4.2.0               </span>
<span class="co">#&gt; [53] yaml_2.3.5                    curl_4.3.2                   </span>
<span class="co">#&gt; [55] memoise_2.0.1                 ggplot2_3.3.6                </span>
<span class="co">#&gt; [57] sass_0.4.1                    biomaRt_2.54.0               </span>
<span class="co">#&gt; [59] stringi_1.7.6                 RSQLite_2.2.14               </span>
<span class="co">#&gt; [61] genefilter_1.80.3             BiocVersion_3.16.0           </span>
<span class="co">#&gt; [63] desc_1.4.1                    filelock_1.0.2               </span>
<span class="co">#&gt; [65] rlang_1.0.2                   pkgconfig_2.0.3              </span>
<span class="co">#&gt; [67] systemfonts_1.0.4             bitops_1.0-7                 </span>
<span class="co">#&gt; [69] evaluate_0.15                 lattice_0.20-45              </span>
<span class="co">#&gt; [71] purrr_0.3.4                   bit_4.0.4                    </span>
<span class="co">#&gt; [73] tidyselect_1.1.2              magrittr_2.0.3               </span>
<span class="co">#&gt; [75] bookdown_0.27                 R6_2.5.1                     </span>
<span class="co">#&gt; [77] generics_0.1.2                DelayedArray_0.24.0          </span>
<span class="co">#&gt; [79] DBI_1.1.3                     pillar_1.7.0                 </span>
<span class="co">#&gt; [81] survival_3.3-1                KEGGREST_1.38.0              </span>
<span class="co">#&gt; [83] RCurl_1.98-1.7                tibble_3.1.7                 </span>
<span class="co">#&gt; [85] crayon_1.5.1                  utf8_1.2.2                   </span>
<span class="co">#&gt; [87] rmarkdown_2.14                progress_1.2.2               </span>
<span class="co">#&gt; [89] locfit_1.5-9.5                grid_4.2.0                   </span>
<span class="co">#&gt; [91] blob_1.2.3                    digest_0.6.29                </span>
<span class="co">#&gt; [93] xtable_1.8-4                  httpuv_1.6.5                 </span>
<span class="co">#&gt; [95] textshaping_0.3.6             munsell_0.5.0                </span>
<span class="co">#&gt; [97] bslib_0.3.1</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Ludwig Geistlinger, Robert Gentleman.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.4.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
